\documentclass[prx,twocolumn,english,longbibliography,superscriptaddress]{revtex4-2}
\usepackage{ifthen}
\usepackage{color}
\usepackage{graphics,graphicx,epsfig}
\usepackage{epsf,epstopdf,wrapfig}
\usepackage{amssymb,amsfonts,amsmath}
\usepackage[export]{adjustbox}
\usepackage[english]{babel}
\include{epsf}
\usepackage{ifthen}		
\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }
\newcommand{\gen}{{\text {gen}}}
\newcommand{\post}{{\text {post}}}
\newcommand{\Treg}{{\text {Treg}}}
\newcommand{\Tconv}{{\text {Tconv}}}
\newcommand{\beqn}{\begin{eqnarray}}
\newcommand{\eeqn}{\end{eqnarray}}
\newcommand{\beq}{\begin{equation}}
\newcommand{\eeq}{\end{equation}}
\newcommand{\av}[1]{\langle{#1}\rangle{}}
\newcommand{\abs}[1]{|#1|}
\newcommand{\<}{\langle}
\newcommand{\Q}{{\mathcal{Q}}}

\renewcommand{\>}{\rangle}
\newcommand{\degree}{{}^{\rm o}}
\newcommand{\bs}{ \mbox{\boldmath$\sigma$}}
\renewcommand{\baselinestretch}{1}
\newcommand{\ssection}[1]{\subsection{#1}}
\newcommand{\mref}[1]{Sec.~\ref{#1}}
\definecolor{junglegreen}{rgb}{0.16, 0.67, 0.53}
\definecolor{myrtle}{rgb}{0.13, 0.26, 0.12}
\definecolor{lincolngreen}{rgb}{0.11, 0.35, 0.02}
\definecolor{forestgreen}{rgb}{0.13, 0.55, 0.13}
\newcommand{\threep}{$3^{\,\prime}$\,}
\newcommand{\fivep}{$5^{\,\prime}$\,}
\newcommand{\AW}[1]{{\color{red}#1}}
\newcommand{\TM}{\color{blue}}
\newcommand{\G}{\color{black}}
\newcommand{\GI}{\color{black}}

\newcommand{\AN}[1]{{\color{black}#1}}
\newcommand{\ANO}[1]{{\color{cyan}#1}}
\usepackage[normalem]{ulem}
\renewcommand\sout{\vphantom}%
\newcommand{\rev}{}
\newcommand{\firstequal}{These authors contributed equally.}
\newcommand{\lastequal}{These authors contributed equally. Please send
  correspondance to {awalczak@lpt.ens.fr}, {tmora@lps.ens.fr}, {armita@uw.com}}
\begin{document}

\onecolumngrid
\begin{center}
{\large \bf Variability in the local and global composition of human T-cell receptor repertoires during thymic development across cell types and individuals}\\
\vspace{0.5cm}

Giulio Isacchini$^{1,2,*}$,  Valentin Quiniou$^{3,4}$, H\'el\`ene Vantomme$^{3,4}$, Paul Stys$^3$,  Encarnita Mariotti-Ferandiz$^{3,\dagger}$, David \\ Klatzmann$^{3,4,\dagger}$, Aleksandra M. Walczak$^{2,\dagger,\ddagger}$, Thierry Mora$^{2,\dagger,\ddagger}$, and Armita Nourmohammad$^{1,5,6,7,8,\dagger,\ddagger}$\\
\vspace{0.3cm}
{\it $^1$Max Planck Institute for Dynamics and Self-organization, Am Fa\ss berg 17, 37077 G\"ottingen, Germany\\
$^2$Laboratoire de physique de l'\'ecole normale sup\'erieure (PSL University), CNRS, Sorbonne Universit\'e,\\ and Universit\'e de Paris, 75005 Paris, France\\
$^3$Sorbonne Universit\'e, INSERM, Immunology-Immunopathology-Immunotherapy (i3),\\ F-75005, Paris, France\\
$^4$AP-HP, H\^opital Piti\'e-Salp\^etri\'ere, Biotherapy (CIC-BTi), F-75651, Paris, France\\
$^5$Department of Physics, University of Washington, 3910 15th Avenue Northeast, Seattle,\\ WA 98195, USA\\
$^6$Paul G. Allen School of Computer Science and Engineering, University of Washington,\\ 85 E Stevens Way NE, Seattle, WA 98195, USA\\
$^7$Department of Applied Mathematics, University of Washington, 4182 W Stevens Way NE,\\ Seattle, WA 98105, USA\\
$^8$Fred Hutchinson Cancer Center, 1241 Eastlake Ave E, Seattle, WA 98102, USA\\
}
\end{center}
\vspace{0.2 cm}
\leftskip=0.5in 
\rightskip=0.5in 

{\small
\noindent$*~$Current affiliation: Peter Debye Institute for Soft Matter Physics, Leipzig University, Leipzig, Germany\\
$\dagger~$Co-senior author\\
$\ddagger~$These authors contributed equally. Correspondence should be addressed to: aleksandra.walczak@phys.ens.fr, thierry.mora@phys.ens.fr, and armita@uw.edu.\\}
\vspace{0.5cm}


{The adaptive immune response relies on T cells that combine phenotypic specialization with diversity of T cell receptors (TCRs)} to recognize a wide range of pathogen{\GI s. TCRs are acquired and selected during T cell maturation in the thymus. Characterizing TCR }repertoires across individuals and {\GI T cell } maturation stages is important for {\GI better understanding adaptive immune responses and for developing } new diagnostics and therapies. {\GI Analyzing a dataset of human TCR repertoires from thymocyte subsets, we } find that the variability between individuals generated during {\GI the TCR }V(D)J recombination is maintained through all stages of T cell maturation and differentiation. The {\GI inter-individual variability } of repertoires of the same cell type is of comparable magnitude to the variability across cell types within the same individual. To zoom in on smaller scales than whole repertoires,   {\GI we defined a distance measuring} the relative overlap of locally similar sequences in repertoires. We find that the whole repertoire models correctly predict local similarity networks, suggesting a lack of forbidden T cell receptor sequences. The local measure correlates well with distances calculated using whole repertoire traits and carries information about cell types. \\

\vspace{0.5cm}

\leftskip=0.in 
\rightskip=0.in 
\twocolumngrid


\section{Introduction} 
{\GI The T cell adaptive immune response leverages various cell subsets. CD8+ cells take on mostly a cytotoxic role,  i.e.  killing infected cells. CD4+ cells differentiate into two subsets, regulatory (Treg) and conventional (Tconv) cells. Tconvs acquire effector helper function and help coordinate the immune response upon activation in the periphery. Tregs modulate the immune response by down-regulating the activity and response of different cells in the immune system. 

To perform their distinct functions, CD8+ and CD4+ cells react to different families of the MHC molecule (class I and II, respectively). Therefore, the selective pressures exerted on their associated TCR repertoires are believed to be markedly different. Indeed, previous studies \cite{EMERSON201314, Li2016a, Carter2019,Isacchini2021a} have described differential properties of the CD4 and CD8 repertoires, reporting significant yet limited statistical differences, mostly related to the V and J gene usage. Similarly, small statistical differences have been reported in the repertoires of conventional (Tconv) versus regulatory (Treg) CD4+ T cells \cite{Isacchini2021a,Lagattuta2022}. However, despite a recent study comparing the repertoires of different thymic subsets in mice \cite{Camaglia2022}, it is still unclear how thymic selection shapes the repertoires of distinct T cell subsets, and how these selection forces vary across individuals.}

To recognize the large variety of possible antigens, T cells express a broad diversity of T-cell receptors (TCR) generated by the random rearrangement of their $\alpha$ and $\beta$ chains. Because this process is stochastic, it may generate receptors with undesirable properties, which must be vetted \cite{Yates2014}.
During their development in the thymus, T cells undergo a selection process that promotes receptors with good affinity to the Major Histocompatibility Complex (MHC), to make sure that they have the minimum necessary recognition properties (\textit{positive selection}). At the same time, receptors with {a} too strong affinity to self-peptides presented by the MHC are less likely to be released into the periphery,  to limit an immune response against the self (\textit{negative selection}).
Since the MHC is highly polymorphic \cite{murphy2008janeway}, this process is expected {\GI to be at least in part specific to }each individual.
Understanding how these processes shape the TCR repertoire and characterizing the heterogeneity within and between individuals, is key to {\GI a} better understanding of the fundamental aspects underlying auto-immunity as well as immune disorders.

High-throughput sequencing of TCR repertoires \cite{Hou2016,Georgiou2014,Bolotin2015a,Mcdaniel2016,DeKosky2013a, Turchaninova2013c,Dekosky2014} has made it possible to study differences between different individuals as well as T cell subsets, giving insight into their respective selection processes. T cells differentiate in the thymus into two major subpopulations of cells depending on whether they express the CD4 or CD8 receptor on their surface. 

{Functionally, CD8+ and CD4+ cells take on different roles. CD8+ cells take on a cytotoxic role and they kill infected cells. CD4+ cells differentiate into two subsets, regulatory (Treg) and conventional (Tconv) cells. Tregs modulate the immune response by down-regulating the activity and response of different cells in the immune system. Tconvs acquire effector helper function and help coordinate the immune response upon activation in the periphery.} 

Because TCR repertoires are so diverse, typical samples still show substantial variability, 
 even when they are from the same subset and same individual~\cite{BensoudaKoraichiPNAS2023}. Comparisons between repertoires must be done at an aggregate level, by identifying statistical features that discriminate between them. A general strategy for this task is to define generative models of TCR sequences for each subset, and compare them using measures of divergence between their distributions \cite{Sethna2020,Isacchini2021a}. We refer to this approach as {\em global} repertoire comparison.
A previous study  \cite{Sethna2020} of bulk TCR repertoires from blood samples showed only moderate variability across individuals, largely driven by differences in the statistics of the recombination process~\cite{Tonegawa1983,Davis1988} rather than individual-specific selection, {which contrasts with the idea that HLA drives the form of the repertoires}.
It remains unclear to what extent  the population variability in the recombination process impacts 
our ability to detect  the  differences between  T cell subsets due to functional selection.

Model-based comparisons, even when they rely on powerful neural networks, are usually dominated by some key statistics of the data, such as germline gene usage, or CDR3 length and position-dependent amino-acid usage. A possible concern is that they can miss  important differences in repertoire properties, such as the depletion or enrichment of particular sequence motifs, which would strongly affect the structure of the repertoires locally in sequence space but may only have a limited impact on their global features as captured by the generative models. 
Previous studies have used local network measures of similarity between closeby sequences to detect immune stimuli or phenotypes shared by TCRs \cite{Pogorelyy2018b,TCRNET,ALICETCRNET,Dash2017,Blackwell2021,Glanville2017,Huang2020}, suggesting that local differences may be important for discriminating between repertoires beyond global statistical features. We refer to such approaches as {\em local} repertoire comparisons.

We applied both global and local comparison methods to TCR repertoire data obtained from the thymi of 9 organ donors, sorted into functional T cell subsets at different levels of thymic maturation. By comparing these different subsets using a combination of model-based and local network analyses, we study how the T cell repertoire evolves across the various stages of thymic development and quantify the heterogeneity between individuals.



% Figure environment removed

\section{Results}

\subsection{Generative models of individual T-cell subsets}

We analyse high-throughput TCR$\beta$ repertoire sequence data of purified T cell subsets from different stages of maturation (see Fig.~\ref{fig:klatz1}A for a schematic of the different stages). We analysed purified samples of CD4+CD8+CD3+ (double positive: DP), CD3+CD8+ (CD8+),  CD3+CD4+CD25+ (Treg), CD3+CD4+CD25- (Tconv) from thymic samples from $n=9$ individuals. Details on sorting strategies and high-throughput sequencing steps are described in \cite{Klatzmann2020} where part of this dataset was presented. DP cells represent an early stage of development after successful recombination and {\G selection of a receptor, prior to commitment to any functional fate.} The statistics of their repertoire should closely follow that of the recombination process. During selection, cells with higher affinity to one of the two major classes of MHC differentiate into either CD8+ or CD4+ cells {\G (Treg and Tconv)}. 

We used the IGoR software \cite{Marcou2018} to infer a VDJ recombination model (called $P_{\rm gen}$) for each of the n=9 individuals from the unproductive TCR sequences pooled from all 4 subsets. Unproductive sequences are fossil records of unsuccessful recombination events on the second chromosome and {\GI are }thus believed to be free of selection effects that impact productive sequences. To accurately account for selection, we then trained a SONIA model \cite{Sethna2020} on productive sequences for each subset, $P_{\rm DP}, P_{\rm CD8}, P_{\Tconv}, P_{\Treg}$, and each individual ($9\times4$ models in total). This model is built on top of the VDJ recombination model, $P=Q\times P_{\rm gen}$, where $Q$ is a subset-specific selection factor dependent on receptor sequence features, including V-, J- gene usages, and junction length and amino acid composition.
As previously done in \cite{Isacchini2021a}, we can {characterise the diversity of repertoires by evaluating the entropy of the inferred models and estimate the similarities between the repertoires by computing the Jensen-Shannon divergence of the models} (see Fig.~\ref{fig:klatz1}B for a summary of the analysis strategy).


\subsection{Thymic selection reduces sequence diversity by amplifying biases of VDJ recombination}

We first quantified how the effective diversity of the repertoire changed throughout thymic development, by computing the Shannon entropy of each inferred model (Fig.~\ref{fig:klatz1}C, top). This entropy {\G is a measure of {\GI the }diversity of TCRs at different developmental stages and can be thought of as the number of distinct TCR sequences. It  is different from} 
 the raw number or clonality of TCR, both of which are subject to sampling biases. 
Following VDJ recombination, when receptors are well described by the $P_{\gen}$ distribution, the entropy is the highest at around 36 bits, 
 corresponding to an effective sequence space of size $2^{36}\approx 10^{11}$. Entropy is significantly reduced at the DP stage (34 bits $\simeq 10^{10}$ sequences), and even further at the single positive stages of  CD8+, Tconv, and Treg (31 bits $\simeq 10^9$ sequences).

We asked whether this reduction of diversity  results from existing biases in the VDJ recombination process, or is independent of them. The bottom part of Fig.~\ref{fig:klatz1}C shows that entropy reduction is accompanied by an increasing correlation between $P_{\rm gen}$ and $Q$ as maturation progresses, meaning that selection reinforces heterogeneities already present at generation, and thus reduces diversity through the ``rich get richer'' effect. This observation is consistent with reports of a similar correlation in fully matured TCR sampled from blood~\cite{Elhanati2014}. The results in Fig.~\ref{fig:klatz1}C  show the progression of this correlation during thymic selection, with onset as early as the DP stage.

\subsection{Global comparison of repertoires between individuals and thymic maturation stages}

We then estimated the similarity between repertoires at various stages of thymic development and in different individuals, by computing the Jensen-Shannon divergence $D_{JS}$ (an information-theoretic measure of distance between distributions) between the inferred models. The resulting distance matrix
(Fig.~S1A) displays a complex structure, which is better interpreted by projecting repertoires into a distance-preserving low-dimensional map using multidimensional scaling (MDS) \cite{MDS}.
We find that 3 embedding dimensions are sufficient to describe the main properties of the distance matrix (Fig.~S1B). {To effectively visualize these embeddings in 2 dimensions,}  we rotated these 3 main axes  in Fig.~\ref{fig:klatz1}D such that the third dimension best aligns with the identity of individuals, which is tracked across subsets by gray lines. We observe a progression of thymic differentiation along dimensions 1 and 2, with clusters corresponding to each stage, while dimension 3 delineates individuals. {\G Notably, the VJ gene features alone cannot separate the intra-individual cell subsets (Fig.~S6B).}
This result suggests that individual heterogeneity imprinted by initial differences in VDJ recombination remains frozen throughout the maturation process,  in agreement with previous analyses on unsorted repertoires \cite{Sethna2020}.

An alternative way to represent similarities between repertoires is to find a common {\G encoding} space for \textit{sequences} (by using neural networks, as discussed in the Methods) instead of whole repertoires, and then  directly compare the location of different receptor sequences from different datasets (Fig.~S2). While sequence variability is large and no clear separation of subsets is visible at the individual sequence level,  we find the average location of receptor sequences {\G in the encoding space } to follow  the expectations from the differentiation process ({\G Fig.~S2), consistent with the global repertoire analy{\GI s}is in Fig.~\ref{fig:klatz1}D}.

{\G In Figs.~\ref{fig:klatz1}E,~\ref{fig:klatz1}F we observe that }  variability of the same cell type between individuals is quantitatively comparable to that between cell types within a single individual. A more detailed quantification of differences across individuals with fixed cell types (Fig.~\ref{fig:klatz1}E) shows that individual differences are stable across stages of maturation. Differences across cell types (with fixed individuals), shown in Fig.~\ref{fig:klatz1}F, follow the known hierarchy of thymic development, with the recombination model $P_{\rm gen}$ furthest from all subsets, but closer to DP, and single positive stages being equidistant from DP and from each other, with the exception of conventional and regulatory CD4+ cells, which are very similar.
% Figure environment removed
\subsection{Repertoires become concentrated as they mature in the thymus}

While the previous analysis gives a general bird eye's view of differences between repertoires, it is not informative about differences at the local sequence level.
Following~\cite{Klatzmann2020} we asked whether the sequence neighborhood of TCR sequences carried signatures of thymic selection. We define as ``neighbors'' sequences that have the same V and J genes, junction length, and differ by at most one amino acid {\G (Hamming distance of 1)}; see Fig.~\ref{fig:klatz2}A. The larger the overall number of neighbors, the more ``concentrated'' is the repertoire. 
Fig.~\ref{fig:klatz2}B shows the distribution of the number of neighbors of an arbitrary TCR from the repertoire, at different stages of T cell development.
% Figure environment removed
The mean number of neighbors increases during thymic development, indicating that the repertoire gets increasingly concentrated around some preferred sequences.  In other words, selection during thymic development amplifies neighboring sequences within certain regions of the sequence space, while depleting others, which results in repertoires with reduced overall diversity but with  local 
agglomeration of sequences.

An alternative way to measure repertoire concentration is to look at the distribution of distances between any two receptor sequences (Fig.~\ref{fig:klatz2}C). Here, we consider the Hamming distance (number of unmatched amino acids in the junction) between sequences with the same V and J gene and junction length.
Receptors tend to have on average lower Hamming distances following selection, for both CD4+ and CD8+ cells (Fig.~\ref{fig:klatz2}B).  Together, these observations are consistent with the results of \cite{Klatzmann2020} on  the CD8+ T cells obtained from the same datasets, and extend them to CD4+ T cells.

Although we observed that the diversity in repertoires decays with maturation (Fig.~\ref{fig:klatz1}C), this global pruning of sequences did not necessarily imply that the pairwise distances between the receptors in a subset should decay with maturation.
We asked whether this local behaviour could be reproduced by our generative models. We pooled data from all patients for each subset to collect enough sequences to be able to train a soNNia (neural-network based) model~\cite{Isacchini2021a} for the selection factor $Q$ for each maturation stage (see Methods).
We find that both the distributions of {\GI the }number of neighbors and {\GI the }Hamming distances are well reproduced by the models (Fig.~S3), indicating that these local neighborhood differences are well captured despite the models being trained on global features of the repertoires.



We then asked whether the model could predict the number of neighbors $n(x)$ of a particular sequence $x$, and not just their distribution. Assuming that each sequence is drawn at random independently of others, the number of neighbors should be distributed according to a Poisson distribution of mean \cite{Pogorelyy2018b,TCRNET}:
\begin{equation}
n_{\rm pred}(x)=\frac{N_{VJ}}{P(V,J)} \sum_{y \in \Omega(x)} P(y) \simeq \frac{N_{VJ} |\Omega(x)| }{P(V,J)} P(x)
\label{eq:rate}
\end{equation}
where $P(x)$ is the model distribution over sequences $x$, $P(V,J)$ is the probability of picking a particular VJ pair estimated from the model, $N_{VJ}$ is the observed number of sequences with that VJ pair, and $\Omega(x)$ is the set of all potential (not necessarily extant) neighbors of $x$.
The approximation in the second equality relies on the assumption that the probability landscape is smooth, so that the probability of a neighboring sequence is on average the same as the focal sequence.

The accuracy of models can be evaluated by calculating the correlation between $n(x)$ and $n_{\rm pred}(x)$ across all sequences with $>3$ neighbours (Fig.~\ref{fig:klatz2}D). {\GI Fig.~\ref{fig:klatz2}E compares the performance of three models across cell subsets: the naked recombination model $P_{\rm gen}$, the linear (SONIA) and the  neural-network (soNNia) selection models. The performance of the soNNia model reaches the Spearman correlation of $\rho\sim 0.75$ for CD8+ and Tconv subsets. The Treg and DP datasets have smaller sizes and contain fewer nearest neighbours on average (Fig.~S7),  resulting in a nosier comparison to the model, which reduces the absolute correlation. The smooth landscape approximation only moderately degrades predictability (Fig.~\ref{fig:klatz2}F), while showing much faster computation times (by a factor  of 19 times the mean junction length), thanks to the fact that it does not require computing the probability of each neighboring sequence. These observations carry over to the prediction of the number of second neighbors (i.e., sequences with at most 2 amino acid differences) and beyond (Fig.~S4).}


Overall, these results show that the local properties of individual repertoires are well captured by the model and that the probability landscape of finding receptors sequences is relatively smooth as a function of sequence distance.  This implies that we cannot identify forbidden regions in the space of receptor sequences. 

\subsection{Local differences between repertoires}
Next, we asked how the neighborhood structures differed between cell types and individuals at the level of single sequences. For a given sequence $x$, we want to measure differences in the local network structure of nearest neighbors across different repertoires. We quantify these differences by defining a distance  based on the spearman correlation between the number of neighbors of each sequence in the two datasets:
\begin{equation}
D_{\rm L}(D_1,D_2)=\sqrt{1-\rho(n_{D_1}(x),n_{D_2}(x))}
\label{eq:local_distance}
\end{equation}
where $n_{D_1}(x)$ and $n_{D_2}(x)$ are the number of neighbors of sequence $x$ in repertoires $D_1$ and $D_2$.

Compared to the Jensen-Shannon Divergence $D_{JS}$, this distance is based on the local sequence information only, and in principle is model-independent. However, its accuracy strongly depends on sequencing depth, because it relies on counting the number of neighbors for each sequence,  see Fig.~S5A. To overcome that difficulty,  we used  the estimators $n_{\rm pred}$ evaluated with the inferred models of each repertoire to compute the $D_L(D_1,D_2)$.

When applied to all possible pairs of repertoires, the resulting local distance $D_{\rm L}$ correlates well with the global distance between repertoires $D_{JS}$ (Spearman $\rho=0.68$); see Fig.~\ref{fig:klatz3}A. Figs.~\ref{fig:klatz3}B,~C show this local distance between individuals and between cell types. These statistics of local distances closely mirror those of global distance shown in Figs.~\ref{fig:klatz1}E,~F. This means that differences in the local sequence structure of  the repertoire follows global differences captured by the models, and that the global models fully captures the local statistics.



\section{Discussion}

Previous work \cite{EMERSON201314,Li2016a,Carter2019,Isacchini2021a} has characterized global statistical differences between the TRB repertoires of fully mature CD4+ and CD8+ T cells. Here we tracked how these differences emerge during thymic development. We find that the diversity of the repertoire shrinks as a result of thymic selection. This reduction occurs through the concentration of the repertoire around particular regions of the sequence space, consistent with the findings of \cite{Klatzmann2020} on CD8+ repertoires from the same dataset. These favored sequences are typically likely to be generated by VDJ recombination (high $P_{\rm gen}$) even before thymic selection, suggesting that the recombination process has evolved to produce sequences that are likely to survive thymic selection. This idea was first proposed in \cite{Elhanati2014} based on the analysis of peripheral repertoires. Our analysis shows how the process unfolds and amplifies from initial receptor recombination ($P_{\rm gen}$) to the DP stage, and then further to each SP stage. 

Network analyses of repertoires have been successfully applied in a variety of contexts \cite{BenHamo2011,BashfordRogers2013,Madi2017,Pogorelyy2018b, ALICETCRNET,Miho2019,Ronel2021}. 
Following \cite{TCRNET}, we showed that diversity reduction is accompanied with the concentration of the network around the high-degree nodes. Remarkably, our models can accurately predict the changes in these local network structures. Our best-achieving model soNNia uses a nonlinear artificial neural network architecture, allowing it to capture complex interactions. However, we expect the high dimensionality of the sequence space to limit that potential, in particular, if the repertoire landscape is irregular or ``rugged,'' with deep valleys and hills. The usual view of thymic selection is that it should deplete specificities to self-antigens. Since antigen-specific TCRs form clusters in sequence space \cite{Dash2017,Glanville2017}, {\GI negative selection could have translated in the elimination of entire clusters corresponding to these forbidden self-antigen specific TCRs. This in turn should have created valleys and shaped a rugged landscape in the sequence space.}
The success of the model in the face of this potential issue is supported by our observation that the landscape is mostly ``smooth'', as demonstrated by the ability to approximate the probability of a sequence by that of its neighbours.
{\GI This highlights that the elimination of specific TCRs is a complex phenomenon, likely multifactorial, integrating not just specificities but also other parameters that control the level of T cell activation.} Understanding how these observations can be reconciled with the classical view of negative selection \cite{Camaglia2022} remains an interesting question for future work.

We further asked whether the local neighbourhood of a sequence carries information about its cell type. We defined a distance between repertoires based on the similarity of their neighborhood structure, which correlates well with the global, model-based Jensen-Shannon divergence, and recapitulates the hierarchy between the different stages of thymic development. While our neighbor-based distance is in principle model-free, in practice we could only evaluate it reliably using our data-driven generative models because of sampling issues. Nonetheless, our results suggest possible ways to use local network information to compare repertoire subsets, and to study their dependence on cell type, health condition, or age.

Local network structures can be used to detect responding clones during an infection, by looking for sequences with more neighbours than expected in a single repertoire \cite{ALICETCRNET}. Our results on comparing repertoires with a local neighborhood distance suggest that these local differences could also be used to identify sequences that are particularly enriched in one repertoire versus another. This could allow us to define sequences that are characteristic of particular repertoires, and use them to better understand the relationship between cell subset and TCR, with potential applications for diagnosis and phenotyping.


 \section{Methods}


\subsection{Code and data availability}
{\G{The code used to reproduce the figures can be found at \path{https://github.com/statbiophys/Global_and_Local_Variability_2023}}}

\subsection{Multidimensional Scaling}
The objective of Multidimensional scaling is to find a lower dimensional representation of data that preserves the similarity between samples in the dataset.  Given $D$ data points and precomputed distances $d_{ij}$ between points it infers data coordinates $(x_1,\dots,x_D)$ with $x_i\in \mathbb{R}^N$ and $N$ small by minimizing an objective function called stress:
\begin{equation}
Stress(x_1,\dots,x_D)=\sqrt{\sum_{i\neq j}(d_{ij}-||x_i-x_j||_2)^2}
\end{equation}

\subsection{Inference of a representation space for TCR sequences}
In order to better visualize the differences between repertoire subsets from different stages of  maturation in the thymus, we develop a method to map the receptor sequences in a representation space that carries information about selection. We build a feed-forward neural network {\GI that} outputs the selection factor for each subset and has a hidden layer of dimension 2; see Fig S2A for a sketch of the network architecture.  We infer selection factors by maximizing the joint objective
\begin{equation}
\mathcal{L}(\theta)= \sum_{t\in \mathcal{T}} \mathbb{E}_t[-E^t_\theta] -\log \mathbb{E}_{\mathcal{G}}[e^{-E^t_\theta}]
\end{equation}
where $\mathcal G$ is the set of sequences generated by the $P_{\gen}$ model, $\theta$ the model parameters, $E_\theta$ the energy that the model assigns to the sequence, and, with abuse of notation, we identify with $\mathcal{T}$ the cell types of the thymic samples and the corresponding dataset. It is important to clarify that this architecture is not an autoencoder because we are not trying to reconstruct the distribution of sequences, but we are only interested in characterizing the selection factor. For this reason, the representation space will carry information only about selection. We implement the model using the Keras software \cite{keras} and infer its parameters using the RMSprop stochastic gradient descent algorithm \cite{HINTON}.  After inference, we map all sequences to the two-dimensional hidden space.  As it can be seen in Fig.~S2B, we do not observe any clear separation between cell types in this representation space. On the other hand, the averages of the distributions are organized in a clear one-dimensional subspace that follows increasing selection.

\subsection{Smoothness of $P_{\rm post}$}
If we assume that the post-selection probability $P_{\rm post} = P_\gen Q$ does not vary significantly in a local region of sequence space, we can ask whether our estimators can predict the number of observed neighbours $N^{obs}(x)$ of a receptor sequence defined by higher cut-offs in {\GI the }hamming distance. In Fig.~S4A we show that the estimators $P^{\rm lin}_{\rm post}$ (linear selection model), $P^{\rm deep}_{\rm post}$  (deep selection model) and $P_{\rm gen}$ (pre-selection generation model) perform reasonably well also in this regime.  Surprisingly, there is no loss in performance for the threshold at hamming distance 2. For higher cut-offs the average performance decreases and the standard deviation increases. The estimators remain however significantly correlated with $N^{obs}(x)$.

The previous result is consistent with the idea that $P_{\rm post}$ is smooth in the neighbourhood of a given sequence. We are then motivated to push this assumption even further to define alternative estimators for $N^{obs}(x)$. The first approximation we can perform is to assume that all neighbours have similar probability, as it is done in the main text. This estimator is computationally more efficient as it requires only a single evaluation of the probability.

Alternatively, we can assume that  selection factors do not vary considerably within a neighbourhood and  approximate
$P_{\rm post}(nn_x) \simeq P_{\rm gen}(nn_x) Q(x)$ for the  post-selection models.  Since $P_{\rm gen}(nn_x)$  can be efficiently estimated via dynamic programming \cite{Sethna2019}, the resulting estimator turns out to be more efficient than the exact one, yet not as much as $P_{\rm post}(x)$. We compare the two approximations in Fig.~S4B for $P_{\rm post}^{\rm deep}$. Their performance is approximately comparable to the $P_{\rm post}^{\rm lin}(nn_x)$ for the smallest cut-off value. They generically perform better than  $P_{\rm gen}(nn_x)$ for all cut-off values that we tested. As expected the estimator that assumes smoothness only in selection $Q$ slightly outperforms the one that assumes smoothness of the whole probability $P_\post$.

In conclusion, we find that the three best choices for estimating the number of nearest neighbours are, $P^{\rm deep}_{\rm post}(x)$, $P_{\rm gen}(nn_x) Q^{\rm deep}_{\rm post}(x)$ and  $P_{\rm post}^{\rm deep}(nn_x)$. The three estimators have increasing performance but are also more computationally expensive: the choice of which one to use will thus depend on the specific application and the amount of available data.  Since these three estimators perform better at higher cut-offs in hamming distance than the original $P_{\rm gen}(nn_x)$ \cite{Pogorelyy2018b} at the smallest cut-off,  we expect that integrating information from higher cut-offs in hamming will increase the statistical power of enrichment analyses.

\subsection{Global Distance between repertoires $D_{\rm JS}$}
The Jensen-Shannon divergence $D_{\rm JS}$ is a symmetric measure between two probability distributions. It can be used to quantify the difference between any two repertoires defined by the post-selection probabilities $P_{\rm post}^r=\Q^{r}P_\gen^r$ and $P_{\rm post}^{r'}=\Q^{r'}P_\gen^{r'}$: 
\begin{equation}
D_{\rm JS}(r,r')=\frac{1}{2}\Big\langle \log_2 \frac{2\Q^{r}}{\Q^{r} +\Q^{r'}} \Big\rangle_{r}
+ \frac{1}{2} \Big\langle \log_2 \frac{2\Q^{r'}}{\Q^{r} +\Q^{r'}} \Big\rangle_{r'}
\label{eq:djs}
\end{equation}
where $\langle \cdot \rangle_{r}$ denotes the expectation value with respect to $P_{\rm post}^r$. In practice, we estimate $\langle \cdot \rangle_{r}$ as an empirical average by sampling 20,000 sequences from $P_{\rm post}^r$.

\subsection{Local Distance between repertoires $D_{\rm L}$}
The local distance $D_{\rm L}$ between repertoires, as defined in Eq.~\ref{eq:local_distance}, can be directly evaluated on  sequences shared between repertoires by computing the correlation between the observed number of neighbors in the respective datasets. In Fig.~S5A, we evaluate the $D_{\rm L}$  between independent subsamples from the same dataset. We show that for typical repertoire sizes that are found in the literature ($10^4$-$10^5$ unique receptor sequences), the estimator is too noisy to resolve the observed differences between repertoires (Figs.~3B,C), which can be as small as $D_L \sim 0.1~{\rm bits}$.  

In order to overcome this difficulty we evaluate the estimator using the expected number of neighbors $n_{\rm pred}$ computed by the models. We find that $D_{\rm L}$  converges quickly as a function of the number of sequences evaluated; see Fig.~S5B.  In the main analysis we evaluate $D_{\rm L}$ with 200 sequences sampled from each dataset.

As sequencing technologies improve, we expect the model-free estimator to successfully be used to compare the local structure of different repertoires.
As a proof of concept, we compare the model-free version of the $D_{\rm L}$ estimator to the global distance $D_{\rm JS}$ for the biggest datasets present in our data (6 datasets with at least 80,000 unique amino acid sequences) and we are able to reproduce the results of Fig.~3A,  see Fig.~S5C.

\section{Acknowledgement}
The study was supported by the European Research Council COG 724208 (AMW,  TM, GI), the ANR-19-CE45-0018 ``RESP-REP" (AMW,  TM) from the Agence Nationale de la Recherche, the DFG grant CRC 1310 ``Predictability in Evolution" (AMW,  AN, TM, GI), the National Institutes of Health MIRA award (R35 GM142795 to A.N.), the CAREER award from the National Science Foundation (grant No: 2045054 to A.N.), the MPRG funding through the Max Planck Society (AN,  GI), the ERC-Advanced TRiPoD (322856 to D.K.), LabEx Transimmunom (ANR-11-IDEX-0004-02 to D.K.), RHU iMAP (ANR-16-RHUS-0001 to D.K.), the European Research Area Network-cardiovascular diseases (ANR-18-ECVD-0001 to E.M.F.), and iReceptorPlus (H2020 Research and Innovation Programme 825821 to D.K. and E.M.F). This work benefited from equipment and services from the iGenSeq core facility at Institut du Cerveau (ICM, Paris, France) for all the data production. 

%\bibliography{byhand.bib,thierry.bib}
%apsrev4-2.bst 2019-01-14 (MD) hand-edited version of apsrev4-1.bst
%Control: key (0)
%Control: author (8) initials jnrlst
%Control: editor formatted (1) identically to author
%Control: production of article title (0) allowed
%Control: page (0) single
%Control: year (1) truncated
%Control: production of eprint (0) enabled
\begin{thebibliography}{38}%
\makeatletter
\providecommand \@ifxundefined [1]{%
 \@ifx{#1\undefined}
}%
\providecommand \@ifnum [1]{%
 \ifnum #1\expandafter \@firstoftwo
 \else \expandafter \@secondoftwo
 \fi
}%
\providecommand \@ifx [1]{%
 \ifx #1\expandafter \@firstoftwo
 \else \expandafter \@secondoftwo
 \fi
}%
\providecommand \natexlab [1]{#1}%
\providecommand \enquote  [1]{``#1''}%
\providecommand \bibnamefont  [1]{#1}%
\providecommand \bibfnamefont [1]{#1}%
\providecommand \citenamefont [1]{#1}%
\providecommand \href@noop [0]{\@secondoftwo}%
\providecommand \href [0]{\begingroup \@sanitize@url \@href}%
\providecommand \@href[1]{\@@startlink{#1}\@@href}%
\providecommand \@@href[1]{\endgroup#1\@@endlink}%
\providecommand \@sanitize@url [0]{\catcode `\\12\catcode `\$12\catcode
  `\&12\catcode `\#12\catcode `\^12\catcode `\_12\catcode `\%12\relax}%
\providecommand \@@startlink[1]{}%
\providecommand \@@endlink[0]{}%
\providecommand \url  [0]{\begingroup\@sanitize@url \@url }%
\providecommand \@url [1]{\endgroup\@href {#1}{\urlprefix }}%
\providecommand \urlprefix  [0]{URL }%
\providecommand \Eprint [0]{\href }%
\providecommand \doibase [0]{https://doi.org/}%
\providecommand \selectlanguage [0]{\@gobble}%
\providecommand \bibinfo  [0]{\@secondoftwo}%
\providecommand \bibfield  [0]{\@secondoftwo}%
\providecommand \translation [1]{[#1]}%
\providecommand \BibitemOpen [0]{}%
\providecommand \bibitemStop [0]{}%
\providecommand \bibitemNoStop [0]{.\EOS\space}%
\providecommand \EOS [0]{\spacefactor3000\relax}%
\providecommand \BibitemShut  [1]{\csname bibitem#1\endcsname}%
\let\auto@bib@innerbib\@empty
%</preamble>
\bibitem [{\citenamefont {Emerson}\ \emph {et~al.}(2013)\citenamefont
  {Emerson}, \citenamefont {Sherwood}, \citenamefont {Desmarais}, \citenamefont
  {Malhotra}, \citenamefont {Phippard},\ and\ \citenamefont
  {Robins}}]{EMERSON201314}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {R.}~\bibnamefont
  {Emerson}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Sherwood}},
  \bibinfo {author} {\bibfnamefont {C.}~\bibnamefont {Desmarais}}, \bibinfo
  {author} {\bibfnamefont {S.}~\bibnamefont {Malhotra}}, \bibinfo {author}
  {\bibfnamefont {D.}~\bibnamefont {Phippard}},\ and\ \bibinfo {author}
  {\bibfnamefont {H.}~\bibnamefont {Robins}},\ }\bibfield  {title} {\bibinfo
  {title} {Estimating the ratio of cd4+ to cd8+ t cells using high-throughput
  sequence data},\ }\href
  {https://doi.org/https://doi.org/10.1016/j.jim.2013.02.002} {\bibfield
  {journal} {\bibinfo  {journal} {J. Immunol. Methods}\ }\textbf {\bibinfo
  {volume} {391}},\ \bibinfo {pages} {14 } (\bibinfo {year}
  {2013})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Li}\ \emph {et~al.}(2016)\citenamefont {Li},
  \citenamefont {Hiroi}, \citenamefont {Zhang}, \citenamefont {Shi},
  \citenamefont {Chen}, \citenamefont {De}, \citenamefont {Metter},
  \citenamefont {Wood}, \citenamefont {Sharov}, \citenamefont {Milner},
  \citenamefont {Becker}, \citenamefont {Zhan},\ and\ \citenamefont
  {Weng}}]{Li2016a}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {H.~M.}\ \bibnamefont
  {Li}}, \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont {Hiroi}}, \bibinfo
  {author} {\bibfnamefont {Y.}~\bibnamefont {Zhang}}, \bibinfo {author}
  {\bibfnamefont {A.}~\bibnamefont {Shi}}, \bibinfo {author} {\bibfnamefont
  {G.}~\bibnamefont {Chen}}, \bibinfo {author} {\bibfnamefont {S.}~\bibnamefont
  {De}}, \bibinfo {author} {\bibfnamefont {E.~J.}\ \bibnamefont {Metter}},
  \bibinfo {author} {\bibfnamefont {r.}~\bibnamefont {Wood}, \bibfnamefont
  {William~H}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Sharov}},
  \bibinfo {author} {\bibfnamefont {J.~D.}\ \bibnamefont {Milner}}, \bibinfo
  {author} {\bibfnamefont {K.~G.}\ \bibnamefont {Becker}}, \bibinfo {author}
  {\bibfnamefont {M.}~\bibnamefont {Zhan}},\ and\ \bibinfo {author}
  {\bibfnamefont {N.-p.}\ \bibnamefont {Weng}},\ }\bibfield  {title} {\bibinfo
  {title} {Tcr$\beta$ repertoire of cd4+ and cd8+ t cells is distinct in
  richness, distribution, and cdr3 amino acid composition},\ }\href
  {https://doi.org/10.1189/jlb.6A0215-071RR} {\bibfield  {journal} {\bibinfo
  {journal} {J. Leukoc. Biol}\ }\textbf {\bibinfo {volume} {99}},\ \bibinfo
  {pages} {505} (\bibinfo {year} {2016})},\ \bibinfo {note}
  {26394815}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Carter}\ \emph {et~al.}(2019)\citenamefont {Carter},
  \citenamefont {Preall}, \citenamefont {Grigaityte}, \citenamefont
  {Goldfless}, \citenamefont {Jeffery}, \citenamefont {Briggs}, \citenamefont
  {Vigneault},\ and\ \citenamefont {Atwal}}]{Carter2019}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {J.~A.}\ \bibnamefont
  {Carter}}, \bibinfo {author} {\bibfnamefont {J.~B.}\ \bibnamefont {Preall}},
  \bibinfo {author} {\bibfnamefont {K.}~\bibnamefont {Grigaityte}}, \bibinfo
  {author} {\bibfnamefont {S.~J.}\ \bibnamefont {Goldfless}}, \bibinfo {author}
  {\bibfnamefont {E.}~\bibnamefont {Jeffery}}, \bibinfo {author} {\bibfnamefont
  {A.~W.}\ \bibnamefont {Briggs}}, \bibinfo {author} {\bibfnamefont
  {F.}~\bibnamefont {Vigneault}},\ and\ \bibinfo {author} {\bibfnamefont
  {G.~S.}\ \bibnamefont {Atwal}},\ }\bibfield  {title} {\bibinfo {title}
  {Single t cell sequencing demonstrates the functional role of $\alpha$$\beta$
  tcr pairing in cell lineage and antigen specificity},\ }\href
  {https://doi.org/10.3389/fimmu.2019.01516} {\bibfield  {journal} {\bibinfo
  {journal} {Front. Immunol.}\ }\textbf {\bibinfo {volume} {10}},\ \bibinfo
  {pages} {1516} (\bibinfo {year} {2019})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Isacchini}\ \emph {et~al.}(2021)\citenamefont
  {Isacchini}, \citenamefont {Walczak}, \citenamefont {Mora},\ and\
  \citenamefont {Nourmohammad}}]{Isacchini2021a}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {G.}~\bibnamefont
  {Isacchini}}, \bibinfo {author} {\bibfnamefont {A.~M.}\ \bibnamefont
  {Walczak}}, \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont {Mora}},\ and\
  \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Nourmohammad}},\
  }\bibfield  {title} {\bibinfo {title} {Deep generative selection models of t
  and b cell receptor repertoires with sonnia},\ }\bibfield  {journal}
  {\bibinfo  {journal} {Proceedings of the National Academy of Sciences}\
  }\textbf {\bibinfo {volume} {118}},\ \href
  {https://doi.org/10.1073/pnas.2023141118} {10.1073/pnas.2023141118} (\bibinfo
  {year} {2021})\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Lagattuta}\ \emph {et~al.}(2022)\citenamefont
  {Lagattuta}, \citenamefont {Kang}, \citenamefont {Nathan}, \citenamefont
  {Pauken}, \citenamefont {Jonsson}, \citenamefont {Rao}, \citenamefont
  {Sharpe}, \citenamefont {Ishigaki},\ and\ \citenamefont
  {Raychaudhuri}}]{Lagattuta2022}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {K.~A.}\ \bibnamefont
  {Lagattuta}}, \bibinfo {author} {\bibfnamefont {J.~B.}\ \bibnamefont {Kang}},
  \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Nathan}}, \bibinfo
  {author} {\bibfnamefont {K.~E.}\ \bibnamefont {Pauken}}, \bibinfo {author}
  {\bibfnamefont {A.~H.}\ \bibnamefont {Jonsson}}, \bibinfo {author}
  {\bibfnamefont {D.~A.}\ \bibnamefont {Rao}}, \bibinfo {author} {\bibfnamefont
  {A.~H.}\ \bibnamefont {Sharpe}}, \bibinfo {author} {\bibfnamefont
  {K.}~\bibnamefont {Ishigaki}},\ and\ \bibinfo {author} {\bibfnamefont
  {S.}~\bibnamefont {Raychaudhuri}},\ }\bibfield  {title} {\bibinfo {title}
  {Repertoire analyses reveal t cell antigen receptor sequence features that
  influence t cell fate},\ }\href {https://doi.org/10.1038/s41590-022-01129-x}
  {\bibfield  {journal} {\bibinfo  {journal} {Nature Immunology}\ }\textbf
  {\bibinfo {volume} {23}},\ \bibinfo {pages} {446} (\bibinfo {year}
  {2022})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Camaglia}\ \emph {et~al.}(2022)\citenamefont
  {Camaglia}, \citenamefont {Ryvkin}, \citenamefont {Greenstein}, \citenamefont
  {Reich-Zeliger}, \citenamefont {Chain}, \citenamefont {Mora}, \citenamefont
  {Walczak},\ and\ \citenamefont {Friedman}}]{Camaglia2022}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {F.}~\bibnamefont
  {Camaglia}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Ryvkin}},
  \bibinfo {author} {\bibfnamefont {E.}~\bibnamefont {Greenstein}}, \bibinfo
  {author} {\bibfnamefont {S.}~\bibnamefont {Reich-Zeliger}}, \bibinfo {author}
  {\bibfnamefont {B.}~\bibnamefont {Chain}}, \bibinfo {author} {\bibfnamefont
  {T.}~\bibnamefont {Mora}}, \bibinfo {author} {\bibfnamefont {A.~M.}\
  \bibnamefont {Walczak}},\ and\ \bibinfo {author} {\bibfnamefont
  {N.}~\bibnamefont {Friedman}},\ }\bibfield  {title} {\bibinfo {title}
  {Population based selection shapes the t cell receptor repertoire during
  thymic development},\ }\bibfield  {journal} {\bibinfo  {journal} {bioRxiv}\
  }\href {https://doi.org/10.1101/2022.02.14.480309}
  {10.1101/2022.02.14.480309} (\bibinfo {year} {2022})\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Yates}(2014)}]{Yates2014}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {A.~J.}\ \bibnamefont
  {Yates}},\ }\bibfield  {title} {\bibinfo {title} {{Theories and
  quantification of thymic selection}},\ }\href
  {https://doi.org/10.3389/fimmu.2014.00013} {\bibfield  {journal} {\bibinfo
  {journal} {Front. Immunol.}\ }\textbf {\bibinfo {volume} {5}},\ \bibinfo
  {pages} {13} (\bibinfo {year} {2014})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Murphy}\ \emph {et~al.}(2008)\citenamefont {Murphy},
  \citenamefont {Murphy}, \citenamefont {Travers}, \citenamefont {Walport},
  \citenamefont {Janeway}, \citenamefont {Mauri},\ and\ \citenamefont
  {Ehrenstein}}]{murphy2008janeway}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {K.}~\bibnamefont
  {Murphy}}, \bibinfo {author} {\bibfnamefont {K.}~\bibnamefont {Murphy}},
  \bibinfo {author} {\bibfnamefont {P.}~\bibnamefont {Travers}}, \bibinfo
  {author} {\bibfnamefont {M.}~\bibnamefont {Walport}}, \bibinfo {author}
  {\bibfnamefont {C.}~\bibnamefont {Janeway}}, \bibinfo {author} {\bibfnamefont
  {C.}~\bibnamefont {Mauri}},\ and\ \bibinfo {author} {\bibfnamefont
  {M.}~\bibnamefont {Ehrenstein}},\ }\href
  {https://books.google.de/books?id=-SNrAAAAMAAJ} {\emph {\bibinfo {title}
  {Janeway's Immunobiology}}},\ \bibinfo {series} {Janeway's Immunobiology}\
  No.\ \bibinfo {number} {v. 978,Num. 0-4129}\ (\bibinfo  {publisher} {Garland
  Science},\ \bibinfo {year} {2008})\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Hou}\ \emph {et~al.}(2016)\citenamefont {Hou},
  \citenamefont {Wang}, \citenamefont {Ding}, \citenamefont {Xie},\ and\
  \citenamefont {Diao}}]{Hou2016}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {X.-L.}\ \bibnamefont
  {Hou}}, \bibinfo {author} {\bibfnamefont {L.}~\bibnamefont {Wang}}, \bibinfo
  {author} {\bibfnamefont {Y.-L.}\ \bibnamefont {Ding}}, \bibinfo {author}
  {\bibfnamefont {Q.}~\bibnamefont {Xie}},\ and\ \bibinfo {author}
  {\bibfnamefont {H.-Y.}\ \bibnamefont {Diao}},\ }\bibfield  {title} {\bibinfo
  {title} {{Current status and recent advances of next generation sequencing
  techniques in immunological repertoire}},\ }\href
  {https://doi.org/10.1038/gene.2016.9} {\bibfield  {journal} {\bibinfo
  {journal} {Genes Immun.}\ }\textbf {\bibinfo {volume} {17}},\ \bibinfo
  {pages} {153} (\bibinfo {year} {2016})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Georgiou}\ \emph {et~al.}(2014)\citenamefont
  {Georgiou}, \citenamefont {Ippolito}, \citenamefont {Beausang}, \citenamefont
  {Busse}, \citenamefont {Wardemann},\ and\ \citenamefont
  {Quake}}]{Georgiou2014}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {G.}~\bibnamefont
  {Georgiou}}, \bibinfo {author} {\bibfnamefont {G.~C.}\ \bibnamefont
  {Ippolito}}, \bibinfo {author} {\bibfnamefont {J.}~\bibnamefont {Beausang}},
  \bibinfo {author} {\bibfnamefont {C.~E.}\ \bibnamefont {Busse}}, \bibinfo
  {author} {\bibfnamefont {H.}~\bibnamefont {Wardemann}},\ and\ \bibinfo
  {author} {\bibfnamefont {S.~R.}\ \bibnamefont {Quake}},\ }\bibfield  {title}
  {\bibinfo {title} {{The promise and challenge of high-throughput sequencing
  of the antibody repertoire.}},\ }\href {https://doi.org/10.1038/nbt.2782}
  {\bibfield  {journal} {\bibinfo  {journal} {Nat. Biotechnol.}\ }\textbf
  {\bibinfo {volume} {32}},\ \bibinfo {pages} {158} (\bibinfo {year} {2014})},\
  \Eprint {https://arxiv.org/abs/NIHMS150003} {arXiv:NIHMS150003} \BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Bolotin}\ \emph {et~al.}(2015)\citenamefont
  {Bolotin}, \citenamefont {Poslavsky}, \citenamefont {Mitrophanov},
  \citenamefont {Shugay}, \citenamefont {Mamedov}, \citenamefont {Putintseva},\
  and\ \citenamefont {Chudakov}}]{Bolotin2015a}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {D.~A.}\ \bibnamefont
  {Bolotin}}, \bibinfo {author} {\bibfnamefont {S.}~\bibnamefont {Poslavsky}},
  \bibinfo {author} {\bibfnamefont {I.}~\bibnamefont {Mitrophanov}}, \bibinfo
  {author} {\bibfnamefont {M.}~\bibnamefont {Shugay}}, \bibinfo {author}
  {\bibfnamefont {I.~Z.}\ \bibnamefont {Mamedov}}, \bibinfo {author}
  {\bibfnamefont {E.~V.}\ \bibnamefont {Putintseva}},\ and\ \bibinfo {author}
  {\bibfnamefont {D.~M.}\ \bibnamefont {Chudakov}},\ }\bibfield  {title}
  {\bibinfo {title} {{MiXCR: software for comprehensive adaptive immunity
  profiling}},\ }\href {https://doi.org/10.1038/nmeth.3364} {\bibfield
  {journal} {\bibinfo  {journal} {Nat. Methods}\ }\textbf {\bibinfo {volume}
  {12}},\ \bibinfo {pages} {380} (\bibinfo {year} {2015})}\BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Mcdaniel}\ \emph {et~al.}(2016)\citenamefont
  {Mcdaniel}, \citenamefont {DeKosky}, \citenamefont {Tanno}, \citenamefont
  {Ellington},\ and\ \citenamefont {Georgiou}}]{Mcdaniel2016}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {J.~R.}\ \bibnamefont
  {Mcdaniel}}, \bibinfo {author} {\bibfnamefont {B.~J.}\ \bibnamefont
  {DeKosky}}, \bibinfo {author} {\bibfnamefont {H.}~\bibnamefont {Tanno}},
  \bibinfo {author} {\bibfnamefont {A.~D.}\ \bibnamefont {Ellington}},\ and\
  \bibinfo {author} {\bibfnamefont {G.}~\bibnamefont {Georgiou}},\ }\bibfield
  {title} {\bibinfo {title} {{Ultra-high-throughput sequencing of the immune
  receptor repertoire from millions of lymphocytes}},\ }\href
  {https://doi.org/10.1038/nprot.2016.024} {\bibfield  {journal} {\bibinfo
  {journal} {Nat. Protoc.}\ }\textbf {\bibinfo {volume} {11}},\ \bibinfo
  {pages} {429} (\bibinfo {year} {2016})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {DeKosky}\ \emph {et~al.}(2013)\citenamefont
  {DeKosky}, \citenamefont {Ippolito}, \citenamefont {Deschner}, \citenamefont
  {Lavinder}, \citenamefont {Wine}, \citenamefont {Rawlings}, \citenamefont
  {Varadarajan}, \citenamefont {Giesecke}, \citenamefont {D{\"{o}}rner},
  \citenamefont {Andrews}, \citenamefont {Wilson}, \citenamefont
  {Hunicke-Smith}, \citenamefont {Willson}, \citenamefont {Ellington},\ and\
  \citenamefont {Georgiou}}]{DeKosky2013a}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {B.~J.}\ \bibnamefont
  {DeKosky}}, \bibinfo {author} {\bibfnamefont {G.~C.}\ \bibnamefont
  {Ippolito}}, \bibinfo {author} {\bibfnamefont {R.~P.}\ \bibnamefont
  {Deschner}}, \bibinfo {author} {\bibfnamefont {J.~J.}\ \bibnamefont
  {Lavinder}}, \bibinfo {author} {\bibfnamefont {Y.}~\bibnamefont {Wine}},
  \bibinfo {author} {\bibfnamefont {B.~M.}\ \bibnamefont {Rawlings}}, \bibinfo
  {author} {\bibfnamefont {N.}~\bibnamefont {Varadarajan}}, \bibinfo {author}
  {\bibfnamefont {C.}~\bibnamefont {Giesecke}}, \bibinfo {author}
  {\bibfnamefont {T.}~\bibnamefont {D{\"{o}}rner}}, \bibinfo {author}
  {\bibfnamefont {S.~F.}\ \bibnamefont {Andrews}}, \bibinfo {author}
  {\bibfnamefont {P.~C.}\ \bibnamefont {Wilson}}, \bibinfo {author}
  {\bibfnamefont {S.~P.}\ \bibnamefont {Hunicke-Smith}}, \bibinfo {author}
  {\bibfnamefont {C.~G.}\ \bibnamefont {Willson}}, \bibinfo {author}
  {\bibfnamefont {A.~D.}\ \bibnamefont {Ellington}},\ and\ \bibinfo {author}
  {\bibfnamefont {G.}~\bibnamefont {Georgiou}},\ }\bibfield  {title} {\bibinfo
  {title} {{High-throughput sequencing of the paired human immunoglobulin heavy
  and light chain repertoire.}},\ }\href {https://doi.org/10.1038/nbt.2492}
  {\bibfield  {journal} {\bibinfo  {journal} {Nat. Biotechnol.}\ }\textbf
  {\bibinfo {volume} {31}},\ \bibinfo {pages} {166} (\bibinfo {year} {2013})},\
  \Eprint {https://arxiv.org/abs/NIHMS150003} {arXiv:NIHMS150003} \BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Turchaninova}\ \emph {et~al.}(2013)\citenamefont
  {Turchaninova}, \citenamefont {Britanova}, \citenamefont {Bolotin},
  \citenamefont {Shugay}, \citenamefont {Putintseva}, \citenamefont
  {Staroverov}, \citenamefont {Sharonov}, \citenamefont {Shcherbo},
  \citenamefont {Zvyagin}, \citenamefont {Mamedov}, \citenamefont {Linnemann},
  \citenamefont {Schumacher},\ and\ \citenamefont
  {Chudakov}}]{Turchaninova2013c}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {M.~a.}\ \bibnamefont
  {Turchaninova}}, \bibinfo {author} {\bibfnamefont {O.~V.}\ \bibnamefont
  {Britanova}}, \bibinfo {author} {\bibfnamefont {D.~a.}\ \bibnamefont
  {Bolotin}}, \bibinfo {author} {\bibfnamefont {M.}~\bibnamefont {Shugay}},
  \bibinfo {author} {\bibfnamefont {E.~V.}\ \bibnamefont {Putintseva}},
  \bibinfo {author} {\bibfnamefont {D.~B.}\ \bibnamefont {Staroverov}},
  \bibinfo {author} {\bibfnamefont {G.}~\bibnamefont {Sharonov}}, \bibinfo
  {author} {\bibfnamefont {D.}~\bibnamefont {Shcherbo}}, \bibinfo {author}
  {\bibfnamefont {I.~V.}\ \bibnamefont {Zvyagin}}, \bibinfo {author}
  {\bibfnamefont {I.~Z.}\ \bibnamefont {Mamedov}}, \bibinfo {author}
  {\bibfnamefont {C.}~\bibnamefont {Linnemann}}, \bibinfo {author}
  {\bibfnamefont {T.~N.}\ \bibnamefont {Schumacher}},\ and\ \bibinfo {author}
  {\bibfnamefont {D.~M.}\ \bibnamefont {Chudakov}},\ }\bibfield  {title}
  {\bibinfo {title} {{Pairing of T-cell receptor chains via emulsion PCR}},\
  }\href {https://doi.org/10.1002/eji.201343453} {\bibfield  {journal}
  {\bibinfo  {journal} {Eur. J. Immunol.}\ }\textbf {\bibinfo {volume} {43}},\
  \bibinfo {pages} {2507} (\bibinfo {year} {2013})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Dekosky}\ \emph {et~al.}(2014)\citenamefont
  {Dekosky}, \citenamefont {Kojima}, \citenamefont {Rodin}, \citenamefont
  {Charab}, \citenamefont {Ippolito}, \citenamefont {Ellington},\ and\
  \citenamefont {Georgiou}}]{Dekosky2014}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {B.~J.}\ \bibnamefont
  {Dekosky}}, \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont {Kojima}},
  \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Rodin}}, \bibinfo
  {author} {\bibfnamefont {W.}~\bibnamefont {Charab}}, \bibinfo {author}
  {\bibfnamefont {G.~C.}\ \bibnamefont {Ippolito}}, \bibinfo {author}
  {\bibfnamefont {A.~D.}\ \bibnamefont {Ellington}},\ and\ \bibinfo {author}
  {\bibfnamefont {G.}~\bibnamefont {Georgiou}},\ }\bibfield  {title} {\bibinfo
  {title} {{In-depth determination and analysis of the human paired heavy- and
  light-chain antibody repertoire}},\ }\href {https://doi.org/10.1038/nm.3743}
  {\bibfield  {journal} {\bibinfo  {journal} {Nat. Med.}\ }\textbf {\bibinfo
  {volume} {21}},\ \bibinfo {pages} {1} (\bibinfo {year} {2014})}\BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Koraichi}\ \emph {et~al.}(2023)\citenamefont
  {Koraichi}, \citenamefont {Ferri}, \citenamefont {Walczak},\ and\
  \citenamefont {Mora}}]{BensoudaKoraichiPNAS2023}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {M.~B.}\ \bibnamefont
  {Koraichi}}, \bibinfo {author} {\bibfnamefont {S.}~\bibnamefont {Ferri}},
  \bibinfo {author} {\bibfnamefont {A.~M.}\ \bibnamefont {Walczak}},\ and\
  \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont {Mora}},\ }\bibfield
  {title} {\bibinfo {title} {Inferring the t cell repertoire dynamics of
  healthy individuals},\ }\href {https://doi.org/10.1073/pnas.2207516120}
  {\bibfield  {journal} {\bibinfo  {journal} {Proceedings of the National
  Academy of Sciences}\ }\textbf {\bibinfo {volume} {120}},\ \bibinfo {pages}
  {e2207516120} (\bibinfo {year} {2023})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Sethna}\ \emph {et~al.}(2020)\citenamefont {Sethna},
  \citenamefont {Isacchini}, \citenamefont {Dupic}, \citenamefont {Mora},
  \citenamefont {Walczak},\ and\ \citenamefont {Elhanati}}]{Sethna2020}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {Z.}~\bibnamefont
  {Sethna}}, \bibinfo {author} {\bibfnamefont {G.}~\bibnamefont {Isacchini}},
  \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont {Dupic}}, \bibinfo
  {author} {\bibfnamefont {T.}~\bibnamefont {Mora}}, \bibinfo {author}
  {\bibfnamefont {A.~M.}\ \bibnamefont {Walczak}},\ and\ \bibinfo {author}
  {\bibfnamefont {Y.}~\bibnamefont {Elhanati}},\ }\bibfield  {title} {\bibinfo
  {title} {{Population variability in the generation and selection of T-cell
  repertoires}},\ }\href {https://doi.org/10.1371/journal.pcbi.1008394}
  {\bibfield  {journal} {\bibinfo  {journal} {PLOS Comput. Biol.}\ }\textbf
  {\bibinfo {volume} {16}},\ \bibinfo {pages} {e1008394} (\bibinfo {year}
  {2020})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Tonegawa}(1983)}]{Tonegawa1983}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {S.}~\bibnamefont
  {Tonegawa}},\ }\bibfield  {title} {\bibinfo {title} {Somatic generation of
  antibody diversity},\ }\href {https://doi.org/10.1038/302575a0} {\bibfield
  {journal} {\bibinfo  {journal} {Nature}\ }\textbf {\bibinfo {volume} {302}},\
  \bibinfo {pages} {575} (\bibinfo {year} {1983})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Davis}\ and\ \citenamefont
  {Bjorkman}(1988)}]{Davis1988}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {M.~M.}\ \bibnamefont
  {Davis}}\ and\ \bibinfo {author} {\bibfnamefont {P.~J.}\ \bibnamefont
  {Bjorkman}},\ }\bibfield  {title} {\bibinfo {title} {T-cell antigen receptor
  genes and t-cell recognition},\ }\href {https://doi.org/10.1038/334395a0}
  {\bibfield  {journal} {\bibinfo  {journal} {Nature}\ }\textbf {\bibinfo
  {volume} {334}},\ \bibinfo {pages} {395} (\bibinfo {year}
  {1988})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Pogorelyy}\ \emph {et~al.}(2018)\citenamefont
  {Pogorelyy}, \citenamefont {Minervina}, \citenamefont {Shugay}, \citenamefont
  {Chudakov}, \citenamefont {Lebedev}, \citenamefont {Mora},\ and\
  \citenamefont {Walczak}}]{Pogorelyy2018b}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {M.~V.}\ \bibnamefont
  {Pogorelyy}}, \bibinfo {author} {\bibfnamefont {A.~A.}\ \bibnamefont
  {Minervina}}, \bibinfo {author} {\bibfnamefont {M.}~\bibnamefont {Shugay}},
  \bibinfo {author} {\bibfnamefont {D.~M.}\ \bibnamefont {Chudakov}}, \bibinfo
  {author} {\bibfnamefont {Y.~B.}\ \bibnamefont {Lebedev}}, \bibinfo {author}
  {\bibfnamefont {T.}~\bibnamefont {Mora}},\ and\ \bibinfo {author}
  {\bibfnamefont {A.~M.}\ \bibnamefont {Walczak}},\ }\bibfield  {title}
  {\bibinfo {title} {{Detecting T-cell receptors involved in immune responses
  from single repertoire snapshots}},\ }\bibfield  {journal} {\bibinfo
  {journal} {arXiv:1807.08833}\ }\href {https://doi.org/10.1101/375162}
  {10.1101/375162} (\bibinfo {year} {2018}),\ \Eprint
  {https://arxiv.org/abs/1807.08833} {arXiv:1807.08833} \BibitemShut {NoStop}%
\bibitem [{\citenamefont {Ritvo}\ \emph {et~al.}(2018)\citenamefont {Ritvo},
  \citenamefont {Saadawi}, \citenamefont {Barennes}, \citenamefont {Quiniou},
  \citenamefont {Chaara}, \citenamefont {El~Soufi}, \citenamefont {Bonnet},
  \citenamefont {Six}, \citenamefont {Shugay}, \citenamefont
  {Mariotti-Ferrandiz},\ and\ \citenamefont {Klatzmann}}]{TCRNET}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {P.-G.}\ \bibnamefont
  {Ritvo}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Saadawi}},
  \bibinfo {author} {\bibfnamefont {P.}~\bibnamefont {Barennes}}, \bibinfo
  {author} {\bibfnamefont {V.}~\bibnamefont {Quiniou}}, \bibinfo {author}
  {\bibfnamefont {W.}~\bibnamefont {Chaara}}, \bibinfo {author} {\bibfnamefont
  {K.}~\bibnamefont {El~Soufi}}, \bibinfo {author} {\bibfnamefont
  {B.}~\bibnamefont {Bonnet}}, \bibinfo {author} {\bibfnamefont
  {A.}~\bibnamefont {Six}}, \bibinfo {author} {\bibfnamefont {M.}~\bibnamefont
  {Shugay}}, \bibinfo {author} {\bibfnamefont {E.}~\bibnamefont
  {Mariotti-Ferrandiz}},\ and\ \bibinfo {author} {\bibfnamefont
  {D.}~\bibnamefont {Klatzmann}},\ }\bibfield  {title} {\bibinfo {title}
  {High-resolution repertoire analysis reveals a major bystander activation of
  tfh and tfr cells},\ }\href {https://doi.org/10.1073/pnas.1808594115}
  {\bibfield  {journal} {\bibinfo  {journal} {Proceedings of the National
  Academy of Sciences}\ }\textbf {\bibinfo {volume} {115}},\ \bibinfo {pages}
  {9604} (\bibinfo {year} {2018})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Pogorelyy}\ and\ \citenamefont
  {Shugay}(2019)}]{ALICETCRNET}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {M.~V.}\ \bibnamefont
  {Pogorelyy}}\ and\ \bibinfo {author} {\bibfnamefont {M.}~\bibnamefont
  {Shugay}},\ }\bibfield  {title} {\bibinfo {title} {A framework for annotation
  of antigen specificities in high-throughput t-cell repertoire sequencing
  studies},\ }\href {https://doi.org/10.3389/fimmu.2019.02159} {\bibfield
  {journal} {\bibinfo  {journal} {Frontiers in Immunology}\ }\textbf {\bibinfo
  {volume} {10}},\ \bibinfo {pages} {2159} (\bibinfo {year}
  {2019})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Dash}\ \emph {et~al.}(2017)\citenamefont {Dash},
  \citenamefont {Fiore-Gartland}, \citenamefont {Hertz}, \citenamefont {Wang},
  \citenamefont {Sharma}, \citenamefont {Souquette}, \citenamefont {Crawford},
  \citenamefont {Clemens}, \citenamefont {Nguyen}, \citenamefont {Kedzierska},
  \citenamefont {La~Gruta}, \citenamefont {Bradley},\ and\ \citenamefont
  {Thomas}}]{Dash2017}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {P.}~\bibnamefont
  {Dash}}, \bibinfo {author} {\bibfnamefont {A.~J.}\ \bibnamefont
  {Fiore-Gartland}}, \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont
  {Hertz}}, \bibinfo {author} {\bibfnamefont {G.~C.}\ \bibnamefont {Wang}},
  \bibinfo {author} {\bibfnamefont {S.}~\bibnamefont {Sharma}}, \bibinfo
  {author} {\bibfnamefont {A.}~\bibnamefont {Souquette}}, \bibinfo {author}
  {\bibfnamefont {J.~C.}\ \bibnamefont {Crawford}}, \bibinfo {author}
  {\bibfnamefont {E.~B.}\ \bibnamefont {Clemens}}, \bibinfo {author}
  {\bibfnamefont {T.~H.~O.}\ \bibnamefont {Nguyen}}, \bibinfo {author}
  {\bibfnamefont {K.}~\bibnamefont {Kedzierska}}, \bibinfo {author}
  {\bibfnamefont {N.~L.}\ \bibnamefont {La~Gruta}}, \bibinfo {author}
  {\bibfnamefont {P.}~\bibnamefont {Bradley}},\ and\ \bibinfo {author}
  {\bibfnamefont {P.~G.}\ \bibnamefont {Thomas}},\ }\bibfield  {title}
  {\bibinfo {title} {Quantifiable predictive features define epitope-specific t
  cell receptor repertoires},\ }\href {https://doi.org/10.1038/nature22383}
  {\bibfield  {journal} {\bibinfo  {journal} {Nature}\ }\textbf {\bibinfo
  {volume} {547}},\ \bibinfo {pages} {89} (\bibinfo {year} {2017})}\BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Mayer-Blackwell}\ \emph {et~al.}(2021)\citenamefont
  {Mayer-Blackwell}, \citenamefont {Schattgen}, \citenamefont {Cohen-Lavi},
  \citenamefont {Crawford}, \citenamefont {Souquette}, \citenamefont {Gaevert},
  \citenamefont {Hertz}, \citenamefont {Thomas}, \citenamefont {Bradley},\ and\
  \citenamefont {Fiore-Gartland}}]{Blackwell2021}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {K.}~\bibnamefont
  {Mayer-Blackwell}}, \bibinfo {author} {\bibfnamefont {S.}~\bibnamefont
  {Schattgen}}, \bibinfo {author} {\bibfnamefont {L.}~\bibnamefont
  {Cohen-Lavi}}, \bibinfo {author} {\bibfnamefont {J.~C.}\ \bibnamefont
  {Crawford}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Souquette}},
  \bibinfo {author} {\bibfnamefont {J.~A.}\ \bibnamefont {Gaevert}}, \bibinfo
  {author} {\bibfnamefont {T.}~\bibnamefont {Hertz}}, \bibinfo {author}
  {\bibfnamefont {P.~G.}\ \bibnamefont {Thomas}}, \bibinfo {author}
  {\bibfnamefont {P.}~\bibnamefont {Bradley}},\ and\ \bibinfo {author}
  {\bibfnamefont {A.}~\bibnamefont {Fiore-Gartland}},\ }\bibfield  {title}
  {\bibinfo {title} {Tcr meta-clonotypes for biomarker discovery with
  \textit{tcrdist3} enabled identification of public, hla-restricted clusters
  of sars-cov-2 tcrs},\ }\href {https://doi.org/10.7554/eLife.68605} {\bibfield
   {journal} {\bibinfo  {journal} {eLife}\ }\textbf {\bibinfo {volume} {10}},\
  \bibinfo {pages} {e68605} (\bibinfo {year} {2021})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Glanville}\ \emph {et~al.}(2017)\citenamefont
  {Glanville}, \citenamefont {Huang}, \citenamefont {Nau}, \citenamefont
  {Hatton}, \citenamefont {Wagar}, \citenamefont {Rubelt}, \citenamefont {Ji},
  \citenamefont {Han}, \citenamefont {Krams}, \citenamefont {Pettus},
  \citenamefont {Haas}, \citenamefont {Arlehamn}, \citenamefont {Sette},
  \citenamefont {Boyd}, \citenamefont {Scriba}, \citenamefont {Martinez},\ and\
  \citenamefont {Davis}}]{Glanville2017}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {J.}~\bibnamefont
  {Glanville}}, \bibinfo {author} {\bibfnamefont {H.}~\bibnamefont {Huang}},
  \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Nau}}, \bibinfo {author}
  {\bibfnamefont {O.}~\bibnamefont {Hatton}}, \bibinfo {author} {\bibfnamefont
  {L.~E.}\ \bibnamefont {Wagar}}, \bibinfo {author} {\bibfnamefont
  {F.}~\bibnamefont {Rubelt}}, \bibinfo {author} {\bibfnamefont
  {X.}~\bibnamefont {Ji}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont
  {Han}}, \bibinfo {author} {\bibfnamefont {S.~M.}\ \bibnamefont {Krams}},
  \bibinfo {author} {\bibfnamefont {C.}~\bibnamefont {Pettus}}, \bibinfo
  {author} {\bibfnamefont {N.}~\bibnamefont {Haas}}, \bibinfo {author}
  {\bibfnamefont {C.~S.~L.}\ \bibnamefont {Arlehamn}}, \bibinfo {author}
  {\bibfnamefont {A.}~\bibnamefont {Sette}}, \bibinfo {author} {\bibfnamefont
  {S.~D.}\ \bibnamefont {Boyd}}, \bibinfo {author} {\bibfnamefont {T.~J.}\
  \bibnamefont {Scriba}}, \bibinfo {author} {\bibfnamefont {O.~M.}\
  \bibnamefont {Martinez}},\ and\ \bibinfo {author} {\bibfnamefont {M.~M.}\
  \bibnamefont {Davis}},\ }\bibfield  {title} {\bibinfo {title} {{Identifying
  specificity groups in the T cell receptor repertoire}},\ }\href
  {https://doi.org/10.1038/nature22976} {\bibfield  {journal} {\bibinfo
  {journal} {Nature}\ }\textbf {\bibinfo {volume} {547}},\ \bibinfo {pages}
  {94} (\bibinfo {year} {2017})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Huang}\ \emph {et~al.}(2020)\citenamefont {Huang},
  \citenamefont {Wang}, \citenamefont {Rubelt}, \citenamefont {Scriba},\ and\
  \citenamefont {Davis}}]{Huang2020}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {H.}~\bibnamefont
  {Huang}}, \bibinfo {author} {\bibfnamefont {C.}~\bibnamefont {Wang}},
  \bibinfo {author} {\bibfnamefont {F.}~\bibnamefont {Rubelt}}, \bibinfo
  {author} {\bibfnamefont {T.~J.}\ \bibnamefont {Scriba}},\ and\ \bibinfo
  {author} {\bibfnamefont {M.~M.}\ \bibnamefont {Davis}},\ }\bibfield  {title}
  {\bibinfo {title} {Analyzing the mycobacterium tuberculosis immune response
  by t-cell receptor clustering with gliph2 and genome-wide antigen
  screening},\ }\href {https://doi.org/10.1038/s41587-020-0505-4} {\bibfield
  {journal} {\bibinfo  {journal} {Nature Biotechnology}\ }\textbf {\bibinfo
  {volume} {38}},\ \bibinfo {pages} {1194} (\bibinfo {year}
  {2020})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Quiniou}\ \emph {et~al.}(2023)\citenamefont
  {Quiniou}, \citenamefont {Barennes}, \citenamefont {Mhanna}, \citenamefont
  {Stys}, \citenamefont {Vantomme}, \citenamefont {Zhou}, \citenamefont
  {Martina}, \citenamefont {Coatnoan}, \citenamefont {Barbie}, \citenamefont
  {Pham}, \citenamefont {Clémenceau}, \citenamefont {Vie}, \citenamefont
  {Shugay}, \citenamefont {Six}, \citenamefont {Brandao}, \citenamefont
  {Mallone}, \citenamefont {Mariotti-Ferrandiz},\ and\ \citenamefont
  {Klatzmann}}]{Klatzmann2020}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {V.}~\bibnamefont
  {Quiniou}}, \bibinfo {author} {\bibfnamefont {P.}~\bibnamefont {Barennes}},
  \bibinfo {author} {\bibfnamefont {V.}~\bibnamefont {Mhanna}}, \bibinfo
  {author} {\bibfnamefont {P.}~\bibnamefont {Stys}}, \bibinfo {author}
  {\bibfnamefont {H.}~\bibnamefont {Vantomme}}, \bibinfo {author}
  {\bibfnamefont {Z.}~\bibnamefont {Zhou}}, \bibinfo {author} {\bibfnamefont
  {F.}~\bibnamefont {Martina}}, \bibinfo {author} {\bibfnamefont
  {N.}~\bibnamefont {Coatnoan}}, \bibinfo {author} {\bibfnamefont
  {M.}~\bibnamefont {Barbie}}, \bibinfo {author} {\bibfnamefont {H.-P.}\
  \bibnamefont {Pham}}, \bibinfo {author} {\bibfnamefont {B.}~\bibnamefont
  {Clémenceau}}, \bibinfo {author} {\bibfnamefont {H.}~\bibnamefont {Vie}},
  \bibinfo {author} {\bibfnamefont {M.}~\bibnamefont {Shugay}}, \bibinfo
  {author} {\bibfnamefont {A.}~\bibnamefont {Six}}, \bibinfo {author}
  {\bibfnamefont {B.}~\bibnamefont {Brandao}}, \bibinfo {author} {\bibfnamefont
  {R.}~\bibnamefont {Mallone}}, \bibinfo {author} {\bibfnamefont
  {E.}~\bibnamefont {Mariotti-Ferrandiz}},\ and\ \bibinfo {author}
  {\bibfnamefont {D.}~\bibnamefont {Klatzmann}},\ }\bibfield  {title} {\bibinfo
  {title} {Human thymopoiesis produces polyspecific cd8\textsuperscript{+}
  $\alpha/ \beta$ t cells responding to multiple viral antigens},\ }\href
  {https://doi.org/10.7554/eLife.81274} {\bibfield  {journal} {\bibinfo
  {journal} {eLife}\ }\textbf {\bibinfo {volume} {12}},\ \bibinfo {pages}
  {e81274} (\bibinfo {year} {2023})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Marcou}\ \emph {et~al.}(2018)\citenamefont {Marcou},
  \citenamefont {Mora},\ and\ \citenamefont {Walczak}}]{Marcou2018}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {Q.}~\bibnamefont
  {Marcou}}, \bibinfo {author} {\bibfnamefont {T.}~\bibnamefont {Mora}},\ and\
  \bibinfo {author} {\bibfnamefont {A.~M.}\ \bibnamefont {Walczak}},\
  }\bibfield  {title} {\bibinfo {title} {{High-throughput immune repertoire
  analysis with IGoR}},\ }\href {https://doi.org/10.1038/s41467-018-02832-w}
  {\bibfield  {journal} {\bibinfo  {journal} {Nat. Commun.}\ }\textbf {\bibinfo
  {volume} {9}},\ \bibinfo {pages} {561} (\bibinfo {year} {2018})}\BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Elhanati}\ \emph {et~al.}(2014)\citenamefont
  {Elhanati}, \citenamefont {Murugan}, \citenamefont {Callan}, \citenamefont
  {Mora},\ and\ \citenamefont {Walczak}}]{Elhanati2014}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {Y.}~\bibnamefont
  {Elhanati}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Murugan}},
  \bibinfo {author} {\bibfnamefont {C.~G.}\ \bibnamefont {Callan}}, \bibinfo
  {author} {\bibfnamefont {T.}~\bibnamefont {Mora}},\ and\ \bibinfo {author}
  {\bibfnamefont {A.~M.}\ \bibnamefont {Walczak}},\ }\bibfield  {title}
  {\bibinfo {title} {{Quantifying selection in immune receptor repertoires}},\
  }\href {https://doi.org/10.1073/pnas.1409572111} {\bibfield  {journal}
  {\bibinfo  {journal} {Proc. Natl. Acad. Sci.}\ }\textbf {\bibinfo {volume}
  {111}},\ \bibinfo {pages} {9875} (\bibinfo {year} {2014})}\BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {I}\ and\ \citenamefont {P}(1997)}]{MDS}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {B.}~\bibnamefont {I}}\
  and\ \bibinfo {author} {\bibfnamefont {G.}~\bibnamefont {P}},\ }\href
  {https://www.springer.com/gp/book/9780387251509} {\emph {\bibinfo {title}
  {Modern Multidimensional Scaling - Theory and Application}}},\ Series in
  Statistics\ (\bibinfo  {publisher} {Springer},\ \bibinfo {year}
  {1997})\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Ben-Hamo}\ and\ \citenamefont
  {Efroni}(2011)}]{BenHamo2011}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {R.}~\bibnamefont
  {Ben-Hamo}}\ and\ \bibinfo {author} {\bibfnamefont {S.}~\bibnamefont
  {Efroni}},\ }\bibfield  {title} {\bibinfo {title} {The whole-organism heavy
  chain b cell repertoire from zebrafish self-organizes into distinct network
  features},\ }\href {https://doi.org/10.1186/1752-0509-5-27} {\bibfield
  {journal} {\bibinfo  {journal} {BMC Systems Biology}\ }\textbf {\bibinfo
  {volume} {5}},\ \bibinfo {pages} {27} (\bibinfo {year} {2011})}\BibitemShut
  {NoStop}%
\bibitem [{\citenamefont {Bashford-Rogers}\ \emph {et~al.}(2013)\citenamefont
  {Bashford-Rogers}, \citenamefont {Palser}, \citenamefont {Huntly},
  \citenamefont {Rance}, \citenamefont {Vassiliou}, \citenamefont {Follows},\
  and\ \citenamefont {Kellam}}]{BashfordRogers2013}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {R.~J.~M.}\
  \bibnamefont {Bashford-Rogers}}, \bibinfo {author} {\bibfnamefont {A.~L.}\
  \bibnamefont {Palser}}, \bibinfo {author} {\bibfnamefont {B.~J.}\
  \bibnamefont {Huntly}}, \bibinfo {author} {\bibfnamefont {R.}~\bibnamefont
  {Rance}}, \bibinfo {author} {\bibfnamefont {G.~S.}\ \bibnamefont
  {Vassiliou}}, \bibinfo {author} {\bibfnamefont {G.~A.}\ \bibnamefont
  {Follows}},\ and\ \bibinfo {author} {\bibfnamefont {P.}~\bibnamefont
  {Kellam}},\ }\bibfield  {title} {\bibinfo {title} {Network properties derived
  from deep sequencing of human b-cell receptor repertoires delineate b-cell
  populations},\ }\href@noop {} {\bibfield  {journal} {\bibinfo  {journal}
  {Genome Res}\ }\textbf {\bibinfo {volume} {23}},\ \bibinfo {pages} {1874}
  (\bibinfo {year} {2013})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Madi}\ \emph {et~al.}(2017)\citenamefont {Madi},
  \citenamefont {Poran}, \citenamefont {Shifrut}, \citenamefont
  {Reich-Zeliger}, \citenamefont {Greenstein}, \citenamefont {Zaretsky},
  \citenamefont {Arnon}, \citenamefont {Laethem}, \citenamefont {Singer},
  \citenamefont {Lu}, \citenamefont {Sun}, \citenamefont {Cohen},\ and\
  \citenamefont {Friedman}}]{Madi2017}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {A.}~\bibnamefont
  {Madi}}, \bibinfo {author} {\bibfnamefont {A.}~\bibnamefont {Poran}},
  \bibinfo {author} {\bibfnamefont {E.}~\bibnamefont {Shifrut}}, \bibinfo
  {author} {\bibfnamefont {S.}~\bibnamefont {Reich-Zeliger}}, \bibinfo {author}
  {\bibfnamefont {E.}~\bibnamefont {Greenstein}}, \bibinfo {author}
  {\bibfnamefont {I.}~\bibnamefont {Zaretsky}}, \bibinfo {author}
  {\bibfnamefont {T.}~\bibnamefont {Arnon}}, \bibinfo {author} {\bibfnamefont
  {F.~V.}\ \bibnamefont {Laethem}}, \bibinfo {author} {\bibfnamefont
  {A.}~\bibnamefont {Singer}}, \bibinfo {author} {\bibfnamefont
  {J.}~\bibnamefont {Lu}}, \bibinfo {author} {\bibfnamefont {P.~D.}\
  \bibnamefont {Sun}}, \bibinfo {author} {\bibfnamefont {I.~R.}\ \bibnamefont
  {Cohen}},\ and\ \bibinfo {author} {\bibfnamefont {N.}~\bibnamefont
  {Friedman}},\ }\bibfield  {title} {\bibinfo {title} {T cell receptor
  repertoires of mice and humans are clustered in similarity networks around
  conserved public cdr3 sequences},\ }\href
  {https://doi.org/10.7554/eLife.22057} {\bibfield  {journal} {\bibinfo
  {journal} {eLife}\ }\textbf {\bibinfo {volume} {6}},\ \bibinfo {pages}
  {e22057} (\bibinfo {year} {2017})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Miho}\ \emph {et~al.}(2019)\citenamefont {Miho},
  \citenamefont {Ro{\v{s}}kar}, \citenamefont {Greiff},\ and\ \citenamefont
  {Reddy}}]{Miho2019}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {E.}~\bibnamefont
  {Miho}}, \bibinfo {author} {\bibfnamefont {R.}~\bibnamefont {Ro{\v{s}}kar}},
  \bibinfo {author} {\bibfnamefont {V.}~\bibnamefont {Greiff}},\ and\ \bibinfo
  {author} {\bibfnamefont {S.~T.}\ \bibnamefont {Reddy}},\ }\bibfield  {title}
  {\bibinfo {title} {Large-scale network analysis reveals the sequence space
  architecture of antibody repertoires},\ }\href
  {https://doi.org/10.1038/s41467-019-09278-8} {\bibfield  {journal} {\bibinfo
  {journal} {Nat. Commun.}\ }\textbf {\bibinfo {volume} {10}},\ \bibinfo
  {pages} {1321} (\bibinfo {year} {2019})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Ronel}\ \emph {et~al.}(2021)\citenamefont {Ronel},
  \citenamefont {Harries}, \citenamefont {Wicks}, \citenamefont {Oakes},
  \citenamefont {Singleton}, \citenamefont {Dearman}, \citenamefont {Maxwell},\
  and\ \citenamefont {Chain}}]{Ronel2021}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {T.}~\bibnamefont
  {Ronel}}, \bibinfo {author} {\bibfnamefont {M.}~\bibnamefont {Harries}},
  \bibinfo {author} {\bibfnamefont {K.}~\bibnamefont {Wicks}}, \bibinfo
  {author} {\bibfnamefont {T.}~\bibnamefont {Oakes}}, \bibinfo {author}
  {\bibfnamefont {H.}~\bibnamefont {Singleton}}, \bibinfo {author}
  {\bibfnamefont {R.}~\bibnamefont {Dearman}}, \bibinfo {author} {\bibfnamefont
  {G.}~\bibnamefont {Maxwell}},\ and\ \bibinfo {author} {\bibfnamefont
  {B.}~\bibnamefont {Chain}},\ }\bibfield  {title} {\bibinfo {title} {The
  clonal structure and dynamics of the human t cell response to an organic
  chemical hapten},\ }\href {https://doi.org/10.7554/eLife.54747} {\bibfield
  {journal} {\bibinfo  {journal} {eLife}\ }\textbf {\bibinfo {volume} {10}},\
  \bibinfo {pages} {e54747} (\bibinfo {year} {2021})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Chollet}\ \emph {et~al.}(2015)\citenamefont {Chollet}
  \emph {et~al.}}]{keras}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {F.}~\bibnamefont
  {Chollet}} \emph {et~al.},\ }\href {https://github.com/fchollet/keras}
  {\bibinfo {title} {Keras}} (\bibinfo {year} {2015})\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Hinton}(2014)}]{HINTON}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {G.}~\bibnamefont
  {Hinton}},\ }\bibfield  {title} {\bibinfo {title} {Rmsprop},\ }\href
  {http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf} {\
  (\bibinfo {year} {2014})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Sethna}\ \emph {et~al.}(2019)\citenamefont {Sethna},
  \citenamefont {Elhanati}, \citenamefont {Callan}, \citenamefont {Walczak},\
  and\ \citenamefont {Mora}}]{Sethna2019}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {Z.}~\bibnamefont
  {Sethna}}, \bibinfo {author} {\bibfnamefont {Y.}~\bibnamefont {Elhanati}},
  \bibinfo {author} {\bibfnamefont {C.~G.}\ \bibnamefont {Callan}}, \bibinfo
  {author} {\bibfnamefont {A.~M.}\ \bibnamefont {Walczak}},\ and\ \bibinfo
  {author} {\bibfnamefont {T.}~\bibnamefont {Mora}},\ }\bibfield  {title}
  {\bibinfo {title} {{OLGA: fast computation of generation probabilities of B-
  and T-cell receptor amino acid sequences and motifs}},\ }\href
  {https://doi.org/10.1101/367904} {\bibfield  {journal} {\bibinfo  {journal}
  {Bioinformatics}\ }\textbf {\bibinfo {volume} {35}},\ \bibinfo {pages} {2974}
  (\bibinfo {year} {2019})},\ \Eprint {https://arxiv.org/abs/1807.04425}
  {arXiv:1807.04425} \BibitemShut {NoStop}%
\end{thebibliography}%

\renewcommand{\theequation}{S\arabic{equation}}
\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}

\appendix

\onecolumngrid

 % si
\newpage  
 \section{Supplementary Material}

% Figure environment removed
% Figure environment removed

% Figure environment removed

% Figure environment removed

% Figure environment removed

% Figure environment removed

% Figure environment removed


\end{document}
