\documentclass[12pt,a4paper,fleqn]{article}

\usepackage[textheight=23cm,textwidth=15cm]{geometry}
\usepackage{amsmath}
\usepackage[version=3]{mhchem}
\usepackage{graphicx}
\usepackage[american]{babel}
\usepackage{here}
\usepackage[articletitle=true, maxauthors=150]{achemso}
\usepackage{siunitx}
\usepackage{multirow}

\setlength{\parindent}{0mm}
\setlength{\parskip}{2ex}
\linespread{1.1}
\setlength{\jot}{8pt}
\renewcommand\arraystretch{1.5}

\usepackage{url}
\usepackage{xcolor}
\usepackage{lineno}


%%%%%new commands%%%%
\usepackage{xspace}
\begin{document}


\begin{center}

{\LARGE\bf
Automated Preparation of Nanoscopic Structures: Graph-Based Sequence Analysis, Mismatch Detection, and pH-Consistent Protonation with Uncertainty Estimates \\}

\vspace{1cm}

\renewcommand*{\thefootnote}{\fnsymbol{footnote}}


{\large
Katja-Sophia Csizi
and
Markus Reiher\footnote{Corresponding author; e-mail: mreiher@ethz.ch}
}\\[4ex]

\renewcommand*{\thefootnote}{\arabic{footnote}}
\setcounter{footnote}{0}

ETH Zurich, Department of Chemistry and Applied Biosciences, Vladimir-Prelog-Weg 2,
8093 Zurich, Switzerland 

\vspace{.5cm}
July 31, 2023

\vspace{.43cm}

\textbf{Abstract} 
\end{center}
\vspace{-0.5cm}
  
\vspace*{-.41cm}
{\small
Structure and function in nanoscale atomistic assemblies are tightly coupled, and every atom with its specific position and even every electron will have a decisive effect on the electronic structure, and hence, on the molecular properties.
Molecular simulations of nanoscopic atomistic structures therefore require accurately resolved three-dimensional input structures. If extracted from experiment, these structures often suffer from severe uncertainties, of which the lack of information on hydrogen atoms is a prominent example. Hence, experimental structures  require careful review and curation, which is a time-consuming and error-prone process. 
Here, we present a fast and robust protocol for the automated structure analysis, and  pH-consistent protonation, in short, ASAP. For biomolecules as a target, the ASAP protocol integrates sequence analysis and error assessment of a given input structure. ASAP allows for  $\mathrm{pK_a}$\xspace prediction from reference data through Gaussian process regression including uncertainty estimation and connects to system-focused atomistic modeling described in (J. Chem. Theory Comput. 16, 2020, 1646). Although focused on biomolecules, ASAP can be extended to other nanoscopic objects, because most of its design elements rely on a general graph-based 
foundation guaranteeing transferability. The modular character of the underlying pipeline supports different degrees of automation, which allows for (i) efficient feedback loops for human-machine interaction with a low entrance barrier and for (ii) integration into autonomous procedures such as automated force field parametrizations. This facilitates fast switching of the pH-state through on-the-fly system-focused reparametrization during a molecular simulation at virtually no extra computational cost. 
}
\newpage


\section{Introduction}
\label{sec:introduction}
The generation of a high-fidelity initial 
structure of a nanoscopic chemical system for atomistic modeling 
is an important, but nontrivial task. Especially for functional biopolymers, for which extensive experimental structural databases exist, minimal structural rearrangements within an active site or through allosteric effects can change, drive, or even impede chemical processes such as enzyme catalysis.\cite{Wodak2019, Zhou2019}
 
Of special usefulness and need is a protein-oriented framework that enables control and building of structural models in a fast, reproducible, robust, physically realistic, standardized, and automated manner, making it possible to process structures with minimal and unbiased human intervention. Only by such frameworks, computational studies of large structures (for instance, relying on molecular quantum mechanics (QM), molecular mechanics (MM), QM/MM hybrid simulations, or molecular dynamics (MD)) can be efficiently carried out in a comprehensive and reproducible way.\cite{Garcia-Viloca2004} 
While efforts have been made to predict structures from sequence information alone~\cite{Senior2020, Jumper2021, Baek2021}, most simulations rely on experimentally curated input structures taken from X-ray crystallography or NMR spectroscopy, which are collected in the protein data bank (PDB)~\cite{Berman2000}. 

Leveraging experimental data for simulations is appealing and provides dense molecular information such as coordinates, atom types, residue and chain information, connectivity, conformers, and more. However, these structures often suffer from significant uncertainties in the atomic coordinates and other structural errors. For instance, incompletely resolved residues, unreasonable connectivities, and atomic clashes can occur. 
These mismatches must be detected and corrected, which can be handled in an automated fashion for most of the above-mentioned structural issues through implementation of simple chemical rules. For instance, chemical valences of an atom can, in many cases, be evaluated by deriving connectivity information from simple distance-based rules, at least for organic and main-group chemical compounds, which make up for a large part of many nanoscopic bio-macromolecules.

Even accurately resolved experimental data that are not plagued by these mismatches require careful processing. 
For instance, multiple conformers or ensembles of molecular models (determined by NMR experiments)
of the same structure can be condensed into a single PDB file depending on the time resolution of the underlying measurements. For enzymes, the conformers that reflect chemically relevant structural intermediates of a catalytic cycle might be only a small portion of the recorded structures in the file. 
Moreover, proteins may be subjected to modifications (such as mutations) to facilitate the crystallization process,\cite{Longenecker2001} which might require correction for simulations of generic parent systems.

The multitude of the above-mentioned structural uncertainties and mismatches are challenging to detect but propagate into the property prediction in a simulation, because every single atom and its position will have a decisive effect on the electronic structure of the compound. Therefore, it is vital to review carefully every molecular structure prior to simulation. 

Apart from automated mismatch detection, the second challenge of structure evaluation is that crystallography usually lacks resolution of hydrogen atoms (which often can only be recovered by more advanced techniques such as neutron scattering), and the many vacant valences need to be occupied in a 
chemically consistent way, reflecting also chemical constraints such as pH value and solvation. 
Whereas the overall pH-stability of a nanoscopic system is primarily determined by the global $\mathrm{pK_a}$\xspace value of the macromolecule, specific residues adopt site-specific $\mathrm{pK_a}$\xspace values based on their local environment, guiding the protonation state of this site.
However, the assignment of meaningful discrete protonation configurations in a protein is by no means trivial, because the many close and long-range electrostatic interactions that affect the protonation state of one specific site must be accurately captured by the model that is employed to predict the negative logarithm of the acid dissociation constant, the $\mathrm{pK_a}$\xspace. Although the protonated configuration of highest probability could, in principle, be found by extensive search in the space of protonated configurations, this will rapidly become unfeasible because of the combinatorial growth of their number with the number of protonation sites and the number of protons to be added.

In this work, we present a physically meaningful and robust pipeline for \textit{a}utomated \textit{s}tructure \textit{a}nalysis and pH-consistent \textit{p}rotonation (denoted \textit{ASAP}) based on quantum chemical reference data. 
Our pipeline combines sequence analysis of a protein, mismatch detection, addition of hydrogen atoms, solvation, and $\mathrm{pK_a}$\xspace prediction from a quantum reference. It has been implemented as part of our open-source software project \textsc{Swoose}~\cite{swoose1.0.0} within the \textsc{Scine}~\cite{scine} infrastructure. While there are multiple computer programs that can complete individual steps of the structure preparation pipeline, only few unify many of the ASAP features, for instance, the protein preparation wizard\cite{Sastry2013} and the \textsc{PdbFixer} of OpenMM\cite{openMM}.
Compared to them, our ASAP pipeline offers additional features as described in the following.

We accomplish error detection through a graph-isomorphism ansatz that extracts faulty occurrences of structural building blocks within a large structure. We derive protonation probability by a linear regression ansatz learned from a Gaussian process, which is trained on experimental $\mathrm{pK_a}$\xspace values and corresponding quantum chemical reference data. As any physicochemical property model should be equipped with confidence intervals~\cite{Proppe2016, Simm2017}, we provide uncertainty estimates for predicted $\mathrm{pK_a}$\xspace values directly delivered by the Gaussian process, such that special consideration is given to residues whose $\mathrm{pK_a}$\xspace values are close to the system pH because they are more prone to be misclassified. This delivers $\mathrm{pK_a}$\xspace values from the deprotonation free energy in solution, which can be obtained with a quantum chemical method.

Our workflow is seamlessly integrated into our pipeline for automated construction of quantum classical hybrid models.\cite{Brunken2021} This implies that the system-focused atomistic model~\cite{Brunken2020} (SFAM) for the given structure can be generated, and the corresponding quantum region can be selected in an automated fashion.\cite{Brunken2021} 
The design of ASAP also enables automated, on-the-fly (re)-parametrization of the SFAM force field for a specific protonation state configuration, which we will discuss in detail in future work. 
Although our approach is optimized for biomolecular systems, it can be extended to other nanoscopic architectures due to its graph-based and first-principles foundations, both being agnostic to the specific 
atomic composition. 

This work is organized as follows: In section \ref{sec:theory}, we discuss how protonation probabilities can be derived from quantum chemical reference data and corresponding confidence intervals from Gaussian processes, followed by an overview of the computational methodology applied. In section \ref{sec:asap}, we present the conceptual design and all components of the ASAP pipeline.
In section \ref{sec:results}, we demonstrate the structure preparation protocol for chain A of the peptide hormone insulin and subsequently study our approach for a set of protein structures extracted from the PDB. Then, we illustrate that $\mathrm{pK_a}$\xspace values in solution can be derived from a 
linear regression ansatz, for which the isolated amino acids represent an ideal test case as their $\mathrm{pK_a}$\xspace values are well known from experiment. Finally, we derive protonation probabilities for selected sites in chain A of insulin. 


\section{Theoretical Considerations and Computational Methodology}
\label{sec:theory}
\subsection{Protonation Probabilities from a Quantum Reference}
\label{subsec:pka_from_quantum_reference}
The $\mathrm{pK_a}$\xspace value is a quantitative measure of the  propensity of an acidic molecule AH to dissociate into a proton ($\mathrm{H^{+}}$) and the corresponding base ($\mathrm{A^{-}}$) in aqueous solution, measuring the protonation probability at a given pH. It is defined as the negative base-10 logarithm of the acidity constant $K_a$, which is the ratio of the concentrations of proton H$^+$ and base A$^-$ and the acid AH in solution:
\begin{equation}
    pK_a = -\log_{10} (K_a) = -\log_{10}\frac{[A^-][H^+]}{[AH]}.
    \label{eq:equilibrium}
\end{equation}
The acidity constant relates to the Gibbs free energy $\mathrm{\Delta G_a}$ of AH deprotonation 
in solution:
\begin{equation}
    K_a = \exp\Big( - \frac{\Delta G_a}{RT}\Big). 
    \label{eq:ka-gibbs}
\end{equation}
${\Delta G_a}$ in solution is given by\cite{Alongi2010}:
\begin{equation}
 {\Delta G_a = G^{\mathrm{aq}}(A^-) + G^{\mathrm{aq}}(H^+) - G^{\mathrm{aq}}(AH) = \Delta G^{\mathrm{aq}}_0 + G^{\mathrm{aq}}(H^+)} 
 \label{eq:Ga}
\end{equation}
Insertion of Equation (\ref{eq:ka-gibbs}) into Equation (\ref{eq:equilibrium})
yields\cite{Alongi2010}
\begin{equation}
    {pK_a = \frac{\Delta G_a}{R T \ln(10)}}.
    \label{eq:pka}
\end{equation}
For a reliable prediction of $\mathrm{pK_a}$\xspace values, the energy contributions in Equation (\ref{eq:Ga}) must be calculated with sufficient accuracy, because an error of $\mathrm{\sim}$ 1.36 kcal $\mathrm{mol^{-1}}$ in ${\Delta G_a}$
translates into one $\mathrm{pK_a}$\xspace unit of error on the logarithmic scale at $T=$293.15 K.
This accuracy is hard to achieve
for standard quantum chemical methods such as density functional theory (DFT). Moreover, it is difficult to model the free energy of a proton in solution reliably (last term in Equation (\ref{eq:Ga})). Most studies rely on the indirect calculation of ${\Delta G_a}$ from a thermodynamic cycle 
(see, for instance, Refs. \citenum{Ho2009, Casasnovas2014, Ho2015, Seybold2015}). Alternatively, this quantity can be measured in experiments, which, however, can result in a mismatch with the theoretical model chosen for the calculation of the other free energy contributions.

Methods for $\mathrm{pK_a}$\xspace prediction can roughly be grouped into
microscopic, macroscopic, and empirical techniques. 
Microscopic techniques are rooted in grid-based
continuum electrostatic schemes~\cite{Bashford1990, Wang2015a} solving the Poisson--Boltzmann equation. Limitations of the Poisson--Boltzmann ansatz are the neglect of specific protein-ion interactions and that properties are usually calculated for a single structure, which will only be valid if this structure is representative of protonated and deprotonated states.
Moreover, solvation effects described by a dielectric continuum in an implicit solvation model will lack the explicit description of hydrogen bonding between solute and solvent, although they can be crucial.\cite{Ho2014} 
Microscopic methods, by contrast, derive the $\mathrm{pK_a}$\xspace value from quantum chemical calculations,\cite{Klicic2002,Klamt2003, Riojas2014, Ho2015, Thapa2017, Lian2018} all-atom molecular dynamics simulations,\cite{Donnini2011, Itoh2011, Wallace2011, Huang2016a} or quantum classical hybrid models\cite{Li2002, Li2003}. For these methods, the different routes to $\mathrm{pK_a}$\xspace prediction 
build on the calculation of acidity in the gas or solution phase, either with a dielectric continuum or with an explicit solvent model.\cite{Ho2011} Empirical $\mathrm{pK_a}$\xspace prediction is mostly based 
on adding an empirical correction to a baseline model-$\mathrm{pK_a}$\xspace value~\cite{Godoy-Ruiz2005, Li2005, Krieger2006, Bas2008b}. This approach is valuable because of its unmatched readiness, allowing for $\mathrm{pK_a}$\xspace prediction in nanoscale structures within seconds, but it lacks a rigorous physical foundation. Various pipelines implement different routes towards $\mathrm{pK_a}$\xspace prediction (see, for example, \textsc{PypKa}~\cite{Reis2020},\textsc{PDB2PQR},\cite{Dolinsky2004, Dolinsky2007} and \textsc{DelPhi}~\cite{Wang2015a} for Poisson--Boltzmann-based 
$\mathrm{pK_a}$\xspace prediction, the empirical approaches 
${H++}$~\cite{Anandakrishnan2012} and \textsc{Propka},\cite{Olsson2011}, and \textsc{Jaguar}~\cite{Bochevarov2016a}, which is DFT-based). 

Klicic et al. have shown that errors in the computation of terms required in Equation (\ref{eq:pka}) are systematic~\cite{Klicic2002} (see also Refs. \cite{Alongi2010, Ho2014, Ratkova2015}), which can be exploited to leverage $\mathrm{pK_a}$\xspace prediction through a linear regression ansatz. In such an ansatz, the $\mathrm{pK_a}$\xspace value of a given site can be derived from free energies of acid and base only (condensed in ${\Delta G_0^\mathrm{aq}}$ on the right-hand side of Equation (\ref{eq:Ga})). These quantities can be calculated with standard electronic structure methods, whereas the fitting parameters absorb both, the value of $\mathrm{G(H^+)}$ and (ideally) all other deficiencies of the QM method employed: 
\begin{equation}
\begin{aligned}
    {pK_a} & = {\frac{s \Delta G_0^\mathrm{aq}}{ RT \ln(10)} + \frac{s G^\mathrm{aq}(H^+)}{RT \ln(10)}} 
                   =  {k\Delta G_0^\mathrm{aq} + C }.
    \end{aligned}
\label{eq:pka_from_linear_relationship}
\end{equation}
The scaling factor $s$ corresponds to an error correction for some given computational method.\cite{Matsui2012} 
An appropriate training set is then needed to derive the regression parameters $k$ and $C$. 
Not only does the approach then circumvent the need to calculate all quantities in the thermodynamic cycle, but it also delivers an estimate for the free energy of a proton for the applied 
method: 
\begin{equation}
    G^\mathrm{aq}(H^+) = \frac{C}{k}
\end{equation}


\subsection{Confidence Intervals from Gaussian Process Regression}
We apply Gaussian process (GP) regression to the regression problem in 
Equation (\ref{eq:pka_from_linear_relationship}).
GP regression is a probabilistic Bayesian machine learning technique that  predicts continuous quantities from a given prior data set. An important advantage of GP models is that they deliver an estimate for the prediction uncertainty\cite{Pernot2015, Proppe2016, Proppe2017, Simm2016, Simm2017, Proppe2019, Reiher2021}, 
which can be leveraged for physicochemical models provided that suitable measures for molecular similarity can be found (cf., Refs. 
\cite{Simm2018, Proppe2019a, Deringer2021a, Gugler2022}).
GPs are nonparametric models which represent distributions over functions $f(x)$ of data points $X$ that are updated whenever new data becomes available. In a GP, the joint distribution $p(f({x_1}$), ..., $f({x_N}$)) is a multivariate Gaussian distribution $\mathcal{N}$, 
\begin{equation}
 p(\textbf{f} | \textbf{x}) = \mathcal{N}(\textbf{f} | \mathbf{\mu}, \textbf{K}),
\end{equation}
with mean $\mathbf{\mu}$ and covariance matrix $\mathbf{K}$:
\begin{equation}
 K_{ij} = \sigma_f^2 \exp\Big(-\frac{1}{2\theta^2}|\mathbf{x}_i - \mathbf{x}_j|^2\Big)
 \equiv \kappa(\mathbf{x}_i, \mathbf{x}_j).
\end{equation}
The elements of the covariance matrix contain a comparison of two similarity-measure descriptors, ${\textbf{x}_i}$ and ${\textbf{x}_j}$, the variation $\sigma_f^2$, and the smoothness parameter $\theta $. From the prior, a posterior distribution can be derived to make predictions $\mathbf{f}_*$ for new unseen data $\mathbf{x}_*$. The joint distribution of observed values \textbf{f} and \textbf{$\mathbf{f_*}$} is again Gaussian:
\begin{equation}
    p(\mathbf{f_*} | \mathbf{x_*}, \textbf{x}, \textbf{y}) =  \mathcal{N}(\textbf{f}_* | \mathbf{\mathbf{\mu}_*}, \textbf{K}_{*}).
\end{equation}
The predictive mean $\mathbf{\mu}_*$ and predictive covariance matrix $\mathbf{K}_*$ are
 \begin{equation}
\begin{aligned}
    \mathbf{\mu}_* &= \mathrm{\textbf{K}_*^T \textbf{K}_y^{-1}\textbf{y}}, \\
    \mathrm{\textbf{K}_*} &= \mathrm{\textbf{K}_{**} - \textbf{K}_*^T\textbf{K}_y^{-1}\textbf{K}_*},
\end{aligned}     
 \end{equation}
respectively, where $\mathrm{\textbf{K}_y = \textbf{K}+\sigma_y^{2}~\textbf{I}}$ with identity matrix \textbf{I} and noise parameter ${\sigma_y^{2}}$. 
The covariance matrices ${\textbf{K}_*}$ and ${\textbf{K}_{**}}$ define the similarities: ${\textbf{K}_*=\kappa(\textbf{x},\textbf{x}_*)}$ and ${\textbf{K}_{**}=\kappa(\textbf{x}_*,\textbf{x}_*)}$. 

Optimal values of the hyperparameter vector ${\lambda = \{{\theta},\sigma_f^2, \sigma_y^2\}}$ of a GP
for given training data can be derived by maximizing the probability of \textbf{y} given \textbf{x} and the corresponding parameters through the log marginal likelihood, 
\begin{equation}
    \log~p(\mathbf{y}|\mathbf{x}, \lambda) = \log~\mathcal{N}(\mathbf{y}|0,\mathbf{K}_y) = -\frac{1}{2} \mathbf{y}^T \mathbf{K}_y^{-1}\mathbf{y} - \frac{1}{2} \log |\mathbf{K}_y| - \frac{N}{2} \log(2\pi).
\end{equation}
This optimization problem can be solved by a gradient-based optimizer. 
For the regression problem in this work, the data points \textbf{x} comprise the ${\Delta G_0^\mathrm{aq}}$ values and the function values ${f(x)}$ are given by the experimental $\mathrm{pK_a}$\xspace values. 

Although a linear relationship is known for our case, the application of GPs instead of ordinary linear regression provides several advantages here. First of all, the homoscedasticity assumption that the residual terms in ordinary linear regression have the same standard deviation is insufficient for the present problem. If our data point requires extrapolation from the range of training data, the uncertainty of the predictive mean will be high, resulting in an overall heteroscedastic error distribution. Second, GPs offer an easy way to learn (or incorporate) the uncertainty in both our variables \textbf{x} (precisely, the accuracy of the employed QM method to calculate ${\Delta G_0^\mathrm{aq}}$) and our function values \textbf{y} (the accuracy of the experimentally measured $\mathrm{pK_a}$\xspace value). The latter is defined by the hyperparameter ${\sigma_y^2}$. 
Note that Bayesian linear regression and bootstrapped linear least-squares regression~\cite{Proppe2017} are alternatives for the present regression problem. 


\subsection{Computational Methodology}
For all DFT calculations with the \textsc{Orca} program package, version 4.2.0.\cite{Neese2012, Neese2020}
we applied the Perdew--Becke--Ernzerhof (PBE) pure~\cite{Perdew1996a} and the corresponding hybrid PBE0 hybrid~\cite{Ernzerhof1999, Adamo1999} density functionals with an Ahlrich's def2-TZVP basis set.\cite{Schafer1994} The resolution-of-the-identity density-fitting technique was selected to accelerate all PBE calculations.\cite{Eichkorn1995, Eichkorn1997}
Grimme's D3 dispersion correction with Becke--Johnson damping was turned on in all DFT calculations.\cite{Grimme2010a, Grimme2011} 
We applied the implicit solvation model SMD~\cite{Marenich2009} with water as solvent. The dielectric constant was chosen to be 80.4 and the solvent probe radius was set to 1.3 \r{A}. Thermodynamic quantities at 298.15 K were obtained by application of the standard gas-phase model of statistical 
thermodynamics;
that is, by approximating the translational, rotational, and vibrational degrees of freedom by the ideal gas, rigid rotor, and harmonic oscillator models, respectively. 
We applied our \textsc{Molassembler} library~\cite{Sobez2020, Molassembler2023} for all graph-theoretical algorithms. 
As part of this work, GP regression has been made freely available within our open-source \textsc{Scine}~\cite{scine} \textsc{Utilities} library.\cite{Utils2023}
 All timings were measured on a computer equipped with two Intel Xeon E5-2670 CPU with 2.60 GHz base frequency on one single thread.
Structures were visualized using \textsc{PyMol} 2.5.2.\cite{PyMOL}


\section{ASAP: Graph-Based Automated Structure Preparation Pipeline}
\label{sec:asap}
The ASAP pipeline generates accurate structural models from some given structural input data (with the preferred file format being the PDB format).
The preparation is implemented as a sequence of three consecutive steps: In step 1, sequence analysis and error detection are carried out through a subgraph isomorphism ansatz that sorts out potentially faulty parts of the overall structure. In step 2, automated protonation is performed. In step 3, the protonated structure is solvated and the corresponding topology information for subsequent MM parametrization step is generated. 

As part of \textsc{Scine Swoose}\cite{swoose1.0.0}, we provide
both, a fully automated and a semi-automated variant of the ASAP pipeline, which offers high flexibility with regard to the degree of interference between operator and software. In the semi-automated supervised variant, the result of each step is provided as temporary structural output data. These data can be viewed, corrected, and fed back into the pipeline for subsequent steps. The workflow is shown in Figure \ref{fig:flowchart}.  

% Figure environment removed

The building blocks for which the nanoscopic structure is searched must be sorted such that the largest building block is searched for first (to avoid false detection of occurrences in cases where smaller building blocks are elements of larger building blocks, \emph{i.e.}, glycine is a substructure of almost all other amino acids).  

\subsection{Graph-Based Automated Structure Analysis}
\label{subsec:step_1}
In step 1 of the ASAP pipeline, a raw molecular structure is subjected to error detection. 
As our ansatz is currently optimized for proteins, we can carry out an initial sequence analysis at the same time.
Our implementation can straightforwardly be generalized to other nanoscopic systems provided that one defines their atomistic constituents in graph-based form. For instance, metal-organic frameworks may be constructed from their modular entities, which can be defined as molecular graphs (for instance, in an automated way using 
Molassembler\cite{Molassembler2023, Sobez2020}).

The principle idea of our structure analysis algorithm relies on the assumption that large parts of a nanoscopic structure often consist of recurring building blocks. Examples are proteinogenic amino acids in proteins, nucleotides in nucleic acids, triglycerides in lipid double layers, monosaccharides in oligosugar molecules. 

We fragment the input structure into 
a container for building blocks and into a container for all 
substructures that either consist of non--motif structures (containing non-amino-acid structures or amino acids with incorrect connectivity or missing atoms). We denote the resulting containers ``regular module container" (RMC) and ``nonregular module container" (NRMC), respectively. 
If applied to a protein structure, the RMC containers will collect all amino acids that form one or many polypeptide chains 
and the NRMC container will collect all non-amino acid 
structure subunits. 

All building blocks of a nanoscopic structure must be encoded as molecular graphs. Then, occurrences can be efficiently detected by application of graph-isomorphism techniques, that is, by querying the molecular graph of the overall structure for occurrences of each building block, which has been successfully demonstrated for related problems.\cite{Artymiuk1994,Spriggs2003} 
The graph-based approach offers the advantage of a flexible redefinition of the "building block", because any molecular 
structure can be converted to a molecular graph. In this work, our set of building blocks consists of the 20 proteinogenic amino acids, complemented by their selenium-containing analogs of methionine and cysteine (selenomethionine, selenocysteine), and of pyrrolysine. 

For the graph construction, we apply flexible edge (element type) and vertex (topography; in our case: distance-based connectivity) matching criteria, and do not evaluate three-dimensional features of the building blocks (stereochemistry, conformers, and so forth).
Note that we define a chemical bond based on the sum of the relevant van der Waals radii (including an additional threshold of 0.4 \r{A}).
We provide the graph information of each building block as string representations of the corresponding molecular graph. As noted above, our 
software can easily be extended towards other building blocks. For instance, a DNA strand
can be searched for occurrences of the nucleobases by storing string representations of adenine, cytosine, guanine, and thymine. This search algorithm efficiently characterizes the great majority of the input structure, while only a small remainder (the NRMC) 
is to be carefully checked for errors by the operator. This transfers the needle-in-a-haystack search, which comprises an exhaustive manual enumeration of the nanoscopic structure, to a tailored, system-specific and local evaluation of erroneous and nonstandard sites that are buried within a large structure. After finalization of step 1, a temporary output is generated, where the substructures are stored in separated files. 

Moreover, we implemented two additional features into the pipeline: First, if the input structure is given in PDB format, 
multiple states can be separated from one another if identified through a marker in the file (for instance, different conformers or snapshots 
that are condensed in a single file). With this information at hand, one can either process all input structures separately or pick one or a few of them for further preprocessing. Note that we recommend discarding most or all solvent molecules in this step (except those that might contribute to chemical reactivity because they are in the vicinity of the active site), and reinsert them in step 3 through a more systematic explicit cluster-solvation approach (such as the one developed by us in Ref. \citenum{Simm2020}).

Second, we apply a recursive bond search strategy in which we iteratively transfer all correct building blocks that are covalently bound to at least one non-regular module atom (for instance, a metal center) also to the NRMC. 
This can be illustrated with a simple example: Structure processing of plastocyanin (PDB-ID: 1AG6~\cite{Xue1998}) separates the overall structure into 737 atoms in the regular-module polypeptide chain and one single $\mathrm{Cu^{2+}}$ atom stored in NRMC. After recursive bond search, the amino acids $\mathrm{Cys_{84}}$, $\mathrm{His_{37}}$, and $\mathrm{His_{87}}$ are transferred to NRMC, as shown in Figure \ref{fig:connectivity_plastocyanin}. 
The inclusion of the direct coordination environment around the metal site into NRMC
facilitates the assignment of charges and spin states, which is required to generate the SFAM model from QM reference data after structure preparation.

% Figure environment removed


\subsection{Addition of Hydrogen Atoms}
\label{subsec:step_2_and_3}
In step 2 of ASAP, hydrogen atoms (which are usually not resolved in crystallography) are added to both substructures in RMC and NRMC
separately. For this purpose, we exploit the redundancy of the recurring building-block structure motifs, 
which were fully characterized through the graph analysis in the previous step.
In an amino acid, the residue name and the atom type of each atom
together form a unique key from which the protonation pattern can be extracted. We apply the IUPAC-IUB rules for atom typing in amino acids~\cite{iupacAtomtyping}, which is the default format in PDB files.
In this format, atoms are labelled 
according to their distance from the carboxylate carbon atom; \emph{e.g.}, the directly bound C atom is denoted "CA". 
In phenylalanine (PHE), for instance, atom "CG" is a phenylic carbon atom, whereas in lysine (LYS), it is an aliphatic carbon atom, so that the protonation pattern for this atom type is only well-defined in combination with the respective amino acid.
With this information at hand, all atoms in the structure 
can be grouped according to 
how many hydrogen atoms must be added and in which geometric arrangement they should be placed (for the above example, PHE-CG is tetrahedrally coordinated, whereas LYS-CG is trigonal planar). This is schematically shown for
a peptide chain as an example in Figure \ref{fig:atom_typing}. For the 23 amino acids, the above criteria define 
\textit{tetrahedral} groups (with one to three hydrogen atoms to be added),
\textit{bent} groups 
(with one hydrogen atom added), \textit{trigonal planar} groups (with one hydrogen atom added) and \textit{pseudo-tetrahedral} groups (with one or two hydrogen atoms added and at least one polyhedral corner remains vacant). 
Assigning atoms into one of the above groups is trivial at first glance, but it can become complicated if the protonation state of a site depends on the chemical environment (that is, the pH value and the solvent).


The NRMC substructures are protonated by the external library \textsc{OpenBabel}~\cite{OpenBabel}. We emphasize that \textsc{OpenBabel} fills open valences with hydrogen atoms based on a hybridization assignment according to bond lengths and angles, so the result must be inspected with care. 
For this purpose, this step generates temporary output holding the protonated substructures in a human-readable format.

% Figure environment removed


\subsection{Assembly and Solvation}
The third and last step in ASAP reassembles the final atomistic structure from the protonated RMC and NRMC substructures. Once merged, the substructure boundaries are cleaned up, because superfluous hydrogen atoms might have been added as an artifact of the fragmentation. Then, charge information are generated for the full structure. For the RMC, this is realized through application of simple chemical rules that derive atomic charges from the coordination environment of an atom. 
Charges and corresponding spin states for the NRMC need to be provided by the operator. 
Also, topological information about the system's connectivity
are generated, and the final structure is solvated by application of a systematic microsolvation approach 
developed in our group.\cite{Simm2020}
As a success criterion, we evaluate with \textsc{Swoose}
whether the SFAM MM model\cite{Brunken2020} can be generated for the resulting structure: During SFAM model parametrization, the atomistic structure is fragmented, and for each fragment, basic validity checks are carried out (\textit{e.g.}, it is analyzed whether the assigned charge and multiplicity form a fitting pair).  

While the RMC is handled automatically by ASAP and always delivers chemically correct structures (i.e., valences, 
charges, and multiplicities are reasonable and valid), errors in the assembled full system can only be introduced by the operator in the NRMC; for instance, by assigning incorrect charge-multiplicity pairs. For that reason, inspection and manipulation of the NRMC must be carried out carefully. Any introduced errors will later be detected during the parametrization of SFAM through aforementioned validity checks. 


\subsection{Automated $\mathrm{pK_a}$\xspace Prediction from a Quantum Reference}

The assignment of a discrete protonation state for specific sites buried in a large structure as a function of environmental conditions (pH, solvent) is a long-standing problem.
As discussed above, the $\mathrm{pK_a}$\xspace can be predicted from quantities calculated with electronic structure methods, that is the free energy of deprotonation in solution.
The unfavorable scaling of these methods with the number of electrons, however, hampers $\mathrm{pK_a}$\xspace prediction through Equation (\ref{eq:pka}) for nanoscopic structures. 

ASAP implements a parallelizable and physically consistent (i.e., derived from quantum mechanical calculations, typically DFT, instead of 
empirical rules)
$\mathrm{pK_a}$\xspace prediction protocol that is feasible for nanoscale structures by virtue of a divide-and-conquer strategy.
This is achieved by coupling the linear regression procedure reviewed in section \ref{subsec:pka_from_quantum_reference} to the fragmentation scheme developed in the context of the SFAM model parametrization.\cite{Brunken2020} The fragmentation scheme allows for the automated generation of spherical fragments around some central atom, which, in the present context, is a pH-sensitive atom. 
The radius of the sphere defining the size of the generated fragment should be chosen by two criteria: First, the QM reference calculations for the generated fragment must be computationally affordable. Second, the size of the fragment sets a cut-off radius that determines to what extent long-range electrostatic interactions, which affect 
the protonation state, are explicitly evaluated. Every interaction beyond this cut-off radius
will be disregarded, as it is not part of the fragment structure. For each site, both a protonated (AH, acidic) and a corresponding deprotonated 
(A, basic) fragment are automatically constructed, as is shown for one cysteine residue in chain A of insulin in Figure \ref{fig:fragmentation_insulin}. 

% Figure environment removed

The two fragments are subjected to structure optimizations and subsequent Hessian calculations. These reference calculations are carried out during the MM model parametrization step and controlled by the \textsc{Scine} database module~\cite{database1.0.0}. This procedure offers several advantages: First, it exploits the efficient parallelization of the MM model parametrization at negligible additional cost. For example, to generate reference data for the SFAM model of insulin chain A, which consists of 328 atoms, 328 structure optimizations, and 328 subsequent Hessian calculations need to be performed, because a fragment is generated around each atom (the resulting redundancy can be exploited for internal consistency checks). The peptide chain contains only six pH-sensitive sites, that is, 12 additional reference calculations need to be performed for the corresponding second state of the site. Also, the force constants required for MM parametrization can be extracted from these reference calculations for both species. Information about them enables a change of the  pH-value during a molecular simulation 
with fast reparametrization of SFAM without the need to generate additional reference data for a different protonation state. 
In transferable force fields, the parameters can easily be adjusted as the atom type changes upon addition or removal of a proton. In the context of system-focused atomistic models, however, the force field parameters have been optimized for a specific molecular configuration, so that an efficient strategy for parameter re-optimization will be needed if minimal structural rearrangements occur. 


\section{Results and Discussion}
\label{sec:results}
\subsection{Automated Structure Processing of Insulin Chain A}
\label{subsec:automated_processing_insulin}
We selected chain A of the peptide hormone insulin as an example to demonstrate the capabilities of  ASAP. The overall workflow, the resulting structures, and the generated output information are compiled in Figure \ref{fig:processing_insulin}. The starting structure was a hexameric hormone that featured 10 conformers resolved in the experimental PDB file and was taken from PDB-ID 1AI0.\cite{Chang1997}
The zinc active site ($\mathrm{Zn^{2+}}$) coordinates three histidine ligands and one water ligand in a tetrahedral configuration.

We applied step 1 of ASAP to separate the overall structure into its 10 conformers, all being candidates for further preprocessing. In all conformers, 
only two solvent molecules were resolved in the structure close to the active site, and they were therefore kept. After separation of the 10 candidates, we selected candidate 4 for further preprocessing. In all other candidates, 
at least two of the three histidine ligands were
coordinated to the metal site at a distance of at least 2.4 \r{A}, 
which clearly exceeds both the tight range of 1.95-2.10 \r{A} for this type of bonds known from experiment\cite{Zhu1997} and our definition of a chemical bond derived from van der Waals radii (see above).
Note that none of the 10 candidate structures exhibited interatomic distances that fit either one of the two criteria for all Zn-coordinating ligands.

The graph-based analysis in step 1 resulted in a separation into 2380 RMC atoms and 96 NRMC atoms, as depicted in the first gray box in Figure \ref{fig:processing_insulin}. 
The NRMC container was filled with the metal-center active site and the corresponding ligands. In the second step, 
hydrogen atoms were added to the RMC substructure
by applying the clustering strategy based on atom typing and amino acid categorization as described above. All pH-sensitive sites were protonated such that the site is uncharged, by default.
After the addition of hydrogen atoms to the NRMC substructure, we assigned a charge of $+2$ to both zinc atoms (central gray box in Figure \ref{fig:processing_insulin}).
Ultimately, the final processing step 3 assembled both 
substructures and corrected the boundaries (because superfluous hydrogen atoms were added as a result of cutting bonds at the substructure interface), 
and determined the topology by a distance-based connectivity evaluation. All corresponding information was stored in separate files. 
This yielded the final structure depicted in the right gray box in Figure \ref{fig:processing_insulin}).

% Figure environment removed


\subsection{Preparation Benchmark Set and Performance Evaluation}
To investigate the overall efficiency and stability, we applied 
ASAP to three different data sets: The RS126 data set~\cite{Rost1993}, the CB396 data set~\cite{Cuff1999} and the EVA6~\cite{Rost2001} data set. The corresponding PDB identification codes are compiled in Table S1 of the Supporting Information.  
We combined these three data sets to one large test set, which 
consists of, in total, 589 structures. After processing with ASAP, 
we found that only 210 structures required no further preprocessing. 
That is, they consisted exclusively of amino-acid motifs that have a chemically correct composition and connectivity. Overall, 153 structures have at least two conformations resolved. 
For 88 $\%$ of the test-set compounds (518 structures), we found that the NRMC substructure accounted for less than 5 $\%$ in terms of atom count of the full system. Hence, in most cases, only a very small portion of the full structure requires manual inspection and curation.

For completeness, we compared the results obtained with ASAP for selected PDB-IDs to the \textsc{PdbFixer} software provided by OpenMM. \textsc{PdbFixer} also realizes different degrees of automation and precise control of every part of the process. The main difference to ASAP is that the pH-specific protonation states are derived empirically, and that it cannot be directly coupled to the parametrization of a system-focused atomistic model. A more detailed discussion of the structure preparation results of \textsc{PdbFixer} compared to ASAP can be found in the Supporting Information. 

ASAP is computationally efficient and can be performed with negligible cost compared to the SFAM MM parametrization. We 
measured the overall timings for preparation steps 1-3 for the subset of 210 error-free structures.
For the largest structure of this subset 
containing about 13,000 atoms, all three steps could be completed within 71 seconds. 

\subsection{Automated $\mathrm{pK_a}$\xspace Prediction}
\subsubsection{Construction of a Suitable Training Set}

To train the GP model, we assemble six small-molecule data sets for each of the six different
pH-sensitive functional groups that can be found in amino acids (\emph{vide infra}).
These groups 
comprise thiol groups (--SH), alcohol groups  (--OH), phenol groups (Ph--OH), carboxylate groups (--COOH), imidazole groups (Im-H), and primary amines (--NH2).
Once ${\Delta G_0^\mathrm{aq}}$ is calculated for all training-set molecules, the GP must be trained only once for each combination of functional-group class and electronic structure method. 
We selected candidate molecules for the six training sets 
according to two criteria: (i) Accurate experimentally measured $\mathrm{pK_a}$\xspace values should be available for
(ii) preferably small and rigid molecular structure. 
The second criterion mitigates the computational cost for training data evaluation
for different conformers. The training-set molecules and corresponding $\mathrm{pK_a}$\xspace values for phenol and alcohol groups were taken from Ref. \citenum{Thapa2017a}, for thiol groups from Ref. \citenum{Thapa2016}, for imidazole groups from Refs. \citenum{SchmidtAmBusch2004} and \citenum{Matsui2012}, for amines from Refs. \citenum{Matsui2012} and \citenum{Juranic2014}, and for carboxylate groups from Ref. \citenum{Beale2013}; see Figure \ref{fig:gpr_training_molecules}. Their Cartesian coordinates are given in the Supporting Information.

% Figure environment removed

We evaluate the relationship between calculated ${\Delta G_0^\mathrm{aq}}$ and experimental reference $\mathrm{pK_a}$\xspace through comparison of ordinary linear regression and GP regression. As shown in Figure \ref{fig:training_molecules_regression}, the GP efficiently learns the linear relationship from only six data points each. Compared to linear regression, which assumes a homoscedastically distributed error, we find that the uncertainty of the predicted $\mathrm{pK_a}$\xspace from the GP is higher in regions where our training-set data are sparse (and grows even larger outside the boundaries of the training set, which is omitted in Figure \ref{fig:training_molecules_regression}). GP models work well for the given regression problem, because they can be trained on small training data sets. However, the training data presented in this work may be extended to cover an even broader pH region.

We investigated whether semiempirical methods are a viable alternative to conventional DFT, 
as they allow for very fast $\mathrm{pK_a}$\xspace prediction even for large structures at moderate computational cost and have therefore been widely applied to nanoscopic structures.\cite{Kromann2016, Jensen2017, Hunt2020} 
The overall findings strongly indicate that semiempirical methods are likely to be too unreliable for predictions of protonation probabilities (for a detailed discussion, see Supporting Information). 

% Figure environment removed
\clearpage

\subsubsection{$\mathrm{pK_a}$\xspace Prediction For Isolated Amino Acids in Aqueous Solution}
We applied our trained GP model to report $\mathrm{pK_a}$\xspace values of the isolated amino acids glutamic acid (GLU), aspartic acid (ASP), lysine (LYS), tyrosine (TYR), histidine (HIS) and cysteine (CYS) because there are reference $\mathrm{pK_a}$\xspace values available for these amino acids at standard conditions in aqueous solution. We calculated $\mathrm{pK_a}$\xspace values in two ways: 1) free energy difference of acid and base in solution were evaluated according to Equation (\ref{eq:pka_from_linear_relationship}), and 2) electronic energy differences, which do not contain entropy and enthalpy contributions of the nuclear scaffold in the condensed phase, were considered as an efficient 
approximation.
The predicted $\mathrm{pK_a}$\xspace values are compiled in Table \ref{tab:pka_amino_acids}.

\begin{table}[h!]
\centering
\label{tab:pka_amino_acids}
\caption{Predicted $\mathrm{pK_a}$\xspace values of side chains of different amino acids (AA) obtained with GP regression from PBE-D3 and PBE0-D3 
density functionals including D3 dispersion correction compared to literature $\mathrm{pK_a}$\xspace~\cite{litPkasAA} data at standard conditions 
together with the mean absolute error (MAE). The first column for each density functional contains $\mathrm{pK_a}$\xspace values learned from electronic energy differences of acid and base without thermodynamic corrections ($\mathrm{\Delta E_0^\mathrm{aq}}$), while the second column derives this property from Gibbs free energy differences ($\mathrm{\Delta G_0^\mathrm{aq}}$).}
\begin{tabular}{lcrrcrr}
\hline
\multicolumn{1}{c}{\multirow{3}{*}{AA}} &\multicolumn{1}{c}{\multirow{3}{*}{  Ref. $\mathrm{pK_a}$\xspace~\cite{litPkasAA}}} & \multicolumn{5}{c}{predicted $\mathrm{pK_a}$\xspace value}  \\
\cline{3-7}
& & \multicolumn{2}{c}{PBE-D3} & &\multicolumn{2}{c}{PBE0-D3}  \\
& & from ${\Delta E_0^\mathrm{aq}}$ & from ${\Delta G_0^\mathrm{aq}}$ &  &from ${\Delta E_0^\mathrm{aq}}$ & from ${\Delta G_0^\mathrm{aq}}$ \\
\hline
\hline
GLU        &     4.25         & 4.20 $\pm$ 0.26   & 4.74 $\pm$ 0.41 
 &                            & 4.67 $\pm $ 0.38  & 4.86 $\pm$ 0.45 \\
 
ASP         &    3.90         & 4.37 $\pm$ 0.29   & 4.42 $\pm$ 0.31 
 &                            & 4.39 $\pm$ 0.29   & 4.53 $\pm$ 0.34 \\
 
HIS        &     6.04         & 6.52 $\pm$ 0.25   & 6.54 $\pm$ 0.25 
 &                            & 6.04 $\pm$ 0.26   & 6.18 $\pm$ 0.26 \\
 
CYS        &     8.37         & 8.38 $\pm$ 0.29  & 8.47 $\pm$ 0.28 
 &                            & 8.28 $\pm$ 0.30 & 8.49 $\pm$ 0.28 \\

TYR        &     10.07        & 9.73 $\pm$ 0.32   & 9.74 $\pm$ 0.32
 &                            & 9.58 $\pm$ 0.29   & 9.55 $\pm$ 0.29 \\
 
LYS        &     10.54        & 10.07 $\pm$ 0.28  & 10.43 $\pm$ 0.33 
&                             & 9.42 $\pm$ 0.26   & 10.32 $\pm$ 0.32 \\
\hline
\textbf{MAE} & & \multicolumn{1}{c}{0.30} & \multicolumn{1}{c}{0.34} & & \multicolumn{1}{c}{0.43} & \multicolumn{1}{c}{0.37} \\
\hline
\end{tabular}
\end{table}

We found that the predicted $\mathrm{pK_a}$\xspace values for the six amino acids are overall consistent with the experimental values, with an MAE of 0.30 $\mathrm{pK_a}$\xspace units for PBE-D3 and 0.43 $\mathrm{pK_a}$\xspace units for PBE0-D3.
The PBE0-D3 functional slightly overestimated the $\mathrm{pK_a}$\xspace values compared to the PBE-D3 functional with a mean signed error of 0.19 $\mathrm{pK_a}$\xspace units compared to 0.01 $\mathrm{pK_a}$\xspace units. Note that we found basis sets of at least triple-zeta quality to be required for sufficiently reliable predictions. In addition, we found no significant improvement in predicted $\mathrm{pK_a}$\xspace values when free energy differences were evaluated compared to electronic energy differences. Hence, in the following, we chose for $\mathrm{pK_a}$\xspace prediction in nanoscopic structures electronic energy differences only. 

\subsubsection{$\mathrm{pK_a}$\xspace Prediction in Nanoscopic Structures} 
\label{subsec:pka_nanoscale}

We demonstrate $\mathrm{pK_a}$\xspace prediction for nanoscale structures at the example of chain A of the peptide hormone insulin. The initial structure was taken from PDB-ID 1AI0.\cite{Chang1997}  
We calculated the $\mathrm{pK_a}$\xspace value of six pH-sensitive sites in insulin chain A, namely $\mathrm{GLU_{27}}$, $\mathrm{CYS_{48}}$, $\mathrm{TYR_{101}}$, $\mathrm{GLU_{127}}$, $\mathrm{TYR_{147}}$ and $\mathrm{CYS_{153}}$. The subscripts denote the index of the pH-sensitive atoms that were either protonated or deprotonated. As structure optimizations are very costly for structures with more than 300 atoms, we generated different-sized fragments around the pH-sensitive atoms and optimized the structures with PBE-D3/def2-TZVP. The size of the fragment determined the degree to which long-range electrostatic interactions were explicitly considered when evaluating the $\mathrm{pK_a}$\xspace value. 
We report the sensitivity of predicted $\mathrm{pK_a}$\xspace values with respect to system size in Figure \ref{fig:pka_from_spheres_insulin}.

% Figure environment removed

We found that the predicted $\mathrm{pK_a}$\xspace values were very stable with respect to fragment size. This indicates that our workflow presented in section \ref{subsec:pka_nanoscale} is an accurate strategy for predicting the acidity constants in nanoscopic structures. We found that especially carboxylate groups (atom indices 
27 and 127 in the full structure) were very stable over a broad range of differently sized spheres, highlighting that all relevant interactions were accurately modelled even within the smallest sphere of 5.5 \r{A}. Moreover, we found that phenol group 101 was insensitive toward fragment size, and phenol group 147 exhibited slightly increased sensitivity. For thiol groups, we found the largest variation with regard to system size and no convergence of the $\mathrm{pK_a}$\xspace value with increasing fragment size. This observation suggests that the variations in predicted acidity constants are not a consequence of lacking electrostatic interactions, but rather a deficiency of the method or the structure. This issue may be mitigated by application of explicit 
solvent models, more accurate basis sets, and the generation of multiple conformers. 

\section{Conclusions}
Molecular simulation of chemical reactions in nanoscopic environments requires accurate, atomistically resolved molecular structures, because molecular function is tightly coupled to the electronic structure to which every individual atom and electron contribute decisively. However, experimentally determined structures are often subject to significant uncertainties. Hence, structures obtained by experiment must be carefully analyzed and processed before being considered as a starting point for computational modeling. However, this cannot be done manually for routine and high-throughput applications. Our ASAP workflow for structure analysis and protonation achieves automation to a high degree and has been optimized for biomolecular environments. It generates an accurate structural model from
experimental structure information in a fast and efficient manner. Moreover, it combines sequence analysis, error detection, protonation, $\mathrm{pK_a}$\xspace prediction, and solvation in a single 
framework. As a result, very little manual work remains. The ASAP workflow seamlessly integrates into our SFAM hybrid model construction pipeline, where the corresponding system-focused force field and a quantum region can be generated for the obtained structure, which offers a physically consistent protocol for QM/MM modeling. 
        
\section*{Acknowledgements}
The authors gratefully acknowledge financial support through SNF project no. 200021\_182400.
Moreover, K.-S.C. was supported by ETH Zurich and SNF grant no. 200021\_172950-1 awarded to PD Dr. Thomas Hofstetter.

\section*{Supporting Information}
PDB identifiers of the ASAP test set, Cartesian coordinates of the training set for $\mathrm{pK_a}$\xspace prediction, and additional information about the calculation of $\mathrm{pK_a}$\xspace values with semiempirical electronic structure methods, and without implicit solvent models, are provided as additional material in the Supporting Information.

\section*{Data Availability Statement}
ASAP has been implemented in our open-source software \textsc{Scine Swoose}.\cite{swoose1.0.0}
\textsc{Scine Utilities} and \textsc{Scine Molassembler} are available in Refs. \citenum{Utils2023} and \citenum{Molassembler2023}. The \textsc{Scine} framework is also
available from GitHub (\url{https://github.com/qcscine}).


%\bibliography{references.bib}
\providecommand{\latin}[1]{#1}
\makeatletter
\providecommand{\doi}
  {\begingroup\let\do\@makeother\dospecials
  \catcode`\{=1 \catcode`\}=2 \doi@aux}
\providecommand{\doi@aux}[1]{\endgroup\texttt{#1}}
\makeatother
\providecommand*\mcitethebibliography{\thebibliography}
\csname @ifundefined\endcsname{endmcitethebibliography}
  {\let\endmcitethebibliography\endthebibliography}{}
\begin{mcitethebibliography}{94}
\providecommand*\natexlab[1]{#1}
\providecommand*\mciteSetBstSublistMode[1]{}
\providecommand*\mciteSetBstMaxWidthForm[2]{}
\providecommand*\mciteBstWouldAddEndPuncttrue
  {\def\EndOfBibitem{\unskip.}}
\providecommand*\mciteBstWouldAddEndPunctfalse
  {\let\EndOfBibitem\relax}
\providecommand*\mciteSetBstMidEndSepPunct[3]{}
\providecommand*\mciteSetBstSublistLabelBeginEnd[3]{}
\providecommand*\EndOfBibitem{}
\mciteSetBstSublistMode{f}
\mciteSetBstMaxWidthForm{subitem}{(\alph{mcitesubitemcount})}
\mciteSetBstSublistLabelBeginEnd
  {\mcitemaxwidthsubitemform\space}
  {\relax}
  {\relax}

\bibitem[Wodak \latin{et~al.}(2019)Wodak, Paci, Dokholyan, Berezovsky,
  Horovitz, Li, Hilser, Bahar, Karanicolas, Stock, Hamm, Stote, Eberhardt,
  Chebaro, Dejaegere, Cecchini, Changeux, Bolhuis, Vreede, Faccioli, Orioli,
  Ravasio, Yan, Brito, Wyart, Gkeka, Rivalta, Palermo, McCammon,
  Panecka-Hofman, Wade, {Di Pizio}, Niv, Nussinov, Tsai, Jang, Padhorny,
  Kozakov, and McLeish]{Wodak2019}
Wodak,~S.~J.; Paci,~E.; Dokholyan,~N.~V.; Berezovsky,~I.~N.; Horovitz,~A.;
  Li,~J.; Hilser,~V.~J.; Bahar,~I.; Karanicolas,~J.; Stock,~G.; Hamm,~P.;
  Stote,~R.~H.; Eberhardt,~J.; Chebaro,~Y.; Dejaegere,~A.; Cecchini,~M.;
  Changeux,~J.-P.; Bolhuis,~P.~G.; Vreede,~J.; Faccioli,~P.; Orioli,~S.;
  Ravasio,~R.; Yan,~L.; Brito,~C.; Wyart,~M.; Gkeka,~P.; Rivalta,~I.;
  Palermo,~G.; McCammon,~J.~A.; Panecka-Hofman,~J.; Wade,~R.~C.; {Di
  Pizio},~A.; Niv,~M.~Y.; Nussinov,~R.; Tsai,~C.-J.; Jang,~H.; Padhorny,~D.;
  Kozakov,~D.; McLeish,~T. Allostery in Its Many Disguises: From Theory to
  Applications. \emph{Structure} \textbf{2019}, \emph{27}, 566--578\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Zhou \latin{et~al.}(2019)Zhou, Musalgaonkar, Johnson, and
  Taylor]{Zhou2019}
Zhou,~Y.; Musalgaonkar,~S.; Johnson,~A.~W.; Taylor,~D.~W. {Tightly-orchestrated
  rearrangements govern catalytic center assembly of the ribosome}. \emph{Nat.
  Commun.} \textbf{2019}, \emph{10}, 1--11\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Garcia-Viloca \latin{et~al.}(2004)Garcia-Viloca, Poulsen, Truhlar, and
  Gao]{Garcia-Viloca2004}
Garcia-Viloca,~M.; Poulsen,~T.~D.; Truhlar,~D.~G.; Gao,~J. {Sensitivity of
  molecular dynamics simulations to the choice of the X-ray structure used to
  model an enzymatic reaction}. \emph{Protein Sci.} \textbf{2004}, \emph{13},
  2341--2354\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Senior \latin{et~al.}(2020)Senior, Evans, Jumper, Kirkpatrick, Sifre,
  Green, Qin, {\v{Z}}{\'{i}}dek, Nelson, Bridgland, Penedones, Petersen,
  Simonyan, Crossan, Kohli, Jones, Silver, Kavukcuoglu, and
  Hassabis]{Senior2020}
Senior,~A.~W.; Evans,~R.; Jumper,~J.; Kirkpatrick,~J.; Sifre,~L.; Green,~T.;
  Qin,~C.; {\v{Z}}{\'{i}}dek,~A.; Nelson,~A.~W.; Bridgland,~A.; Penedones,~H.;
  Petersen,~S.; Simonyan,~K.; Crossan,~S.; Kohli,~P.; Jones,~D.~T.; Silver,~D.;
  Kavukcuoglu,~K.; Hassabis,~D. {Improved protein structure prediction using
  potentials from deep learning}. \emph{Nature} \textbf{2020}, \emph{577},
  706--710\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Jumper \latin{et~al.}(2021)Jumper, Evans, Pritzel, Green, Figurnov,
  Ronneberger, Tunyasuvunakool, Bates, {\v{Z}}{\'{i}}dek, Potapenko, Bridgland,
  Meyer, Kohl, Ballard, Cowie, Romera-Paredes, Nikolov, Jain, Adler, Back,
  Petersen, Reiman, Clancy, Zielinski, Steinegger, Pacholska, Berghammer,
  Bodenstein, Silver, Vinyals, Senior, Kavukcuoglu, Kohli, and
  Hassabis]{Jumper2021}
Jumper,~J.; Evans,~R.; Pritzel,~A.; Green,~T.; Figurnov,~M.; Ronneberger,~O.;
  Tunyasuvunakool,~K.; Bates,~R.; {\v{Z}}{\'{i}}dek,~A.; Potapenko,~A.;
  Bridgland,~A.; Meyer,~C.; Kohl,~S.~A.; Ballard,~A.~J.; Cowie,~A.;
  Romera-Paredes,~B.; Nikolov,~S.; Jain,~R.; Adler,~J.; Back,~T.; Petersen,~S.;
  Reiman,~D.; Clancy,~E.; Zielinski,~M.; Steinegger,~M.; Pacholska,~M.;
  Berghammer,~T.; Bodenstein,~S.; Silver,~D.; Vinyals,~O.; Senior,~A.~W.;
  Kavukcuoglu,~K.; Kohli,~P.; Hassabis,~D. {Highly accurate protein structure
  prediction with AlphaFold}. \emph{Nature} \textbf{2021}, \emph{596},
  583--589\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Baek \latin{et~al.}(2021)Baek, DiMaio, Anishchenko, Dauparas,
  Ovchinnikov, Lee, Wang, Cong, Kinch, Schaeffer, Mill{\'{a}}n, Park, Adams,
  Glassman, DeGiovanni, Pereira, Rodrigues, van Dijk, Ebrecht, Opperman,
  Sagmeister, Buhlheller, Pavkov-Keller, Rathinaswamy, Dalwadi, Yip, Burke,
  Garcia, Grishin, Adams, Read, and Baker]{Baek2021}
Baek,~M.; DiMaio,~F.; Anishchenko,~I.; Dauparas,~J.; Ovchinnikov,~S.;
  Lee,~G.~R.; Wang,~J.; Cong,~Q.; Kinch,~L.~N.; Schaeffer,~R.~D.;
  Mill{\'{a}}n,~C.; Park,~H.; Adams,~C.; Glassman,~C.~R.; DeGiovanni,~A.;
  Pereira,~J.~H.; Rodrigues,~A.~V.; van Dijk,~A.~A.; Ebrecht,~A.~C.;
  Opperman,~D.~J.; Sagmeister,~T.; Buhlheller,~C.; Pavkov-Keller,~T.;
  Rathinaswamy,~M.~K.; Dalwadi,~U.; Yip,~C.~K.; Burke,~J.~E.; Garcia,~K.~C.;
  Grishin,~N.~V.; Adams,~P.~D.; Read,~R.~J.; Baker,~D. {Accurate prediction of
  protein structures and interactions using a three-track neural network}.
  \emph{Science} \textbf{2021}, \emph{373}, 871--876\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Berman(2000)]{Berman2000}
Berman,~H.~M. {The Protein Data Bank}. \emph{Nucleic Acids Res.} \textbf{2000},
  \emph{28}, 235--242\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Longenecker \latin{et~al.}(2001)Longenecker, Garrard, Sheffield, and
  Derewenda]{Longenecker2001}
Longenecker,~K.~L.; Garrard,~S.~M.; Sheffield,~P.~J.; Derewenda,~Z.~S. {Protein
  crystallization by rational mutagenesis of surface residues: Lys to Ala
  mutations promote crystallization of RhoGDI}. \emph{Acta Crystallogr. Sect. D
  Biol. Crystallogr.} \textbf{2001}, \emph{57}, 679--688\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Brunken \latin{et~al.}(2021)Brunken, Csizi, and Reiher]{swoose1.0.0}
Brunken,~C.; Csizi,~K.-S.; Reiher,~M. {qcscine/swoose: Release 1.0.0}.
  2021\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Reiher \latin{et~al.}()Reiher, \latin{et~al.} others]{scine}
Reiher,~M., \latin{et~al.}  {SCINE $-$ Software for Chemical Interaction
  Networks}. \url{http://scine.ethz.ch/}, visited on 2021-10-26\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[{Madhavi Sastry} \latin{et~al.}(2013){Madhavi Sastry}, Adzhigirey,
  Day, Annabhimoju, and Sherman]{Sastry2013}
{Madhavi Sastry},~G.; Adzhigirey,~M.; Day,~T.; Annabhimoju,~R.; Sherman,~W.
  {Protein and ligand preparation: parameters, protocols, and influence on
  virtual screening enrichments}. \emph{J. Comput. Aided. Mol. Des.}
  \textbf{2013}, \emph{27}, 221--234\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Eastman \latin{et~al.}(2017)Eastman, Swails, Chodera, McGibbon, Zhao,
  Beauchamp, Wang, Simmonett, Harrigan, Stern, Wiewiora, Brooks, and
  Pande]{openMM}
Eastman,~P.; Swails,~J.; Chodera,~J.~D.; McGibbon,~R.~T.; Zhao,~Y.;
  Beauchamp,~K.~A.; Wang,~L.-P.; Simmonett,~A.~C.; Harrigan,~M.~P.;
  Stern,~C.~D.; Wiewiora,~R.~P.; Brooks,~B.~R.; Pande,~V.~S. OpenMM 7: Rapid
  development of high performance algorithms for molecular dynamics. \emph{PLOS
  Computational Biology} \textbf{2017}, \emph{13}, 1--17\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Proppe \latin{et~al.}(2016)Proppe, Husch, Simm, and
  Reiher]{Proppe2016}
Proppe,~J.; Husch,~T.; Simm,~G.~N.; Reiher,~M. {Uncertainty quantification for
  quantum chemical models of complex reaction networks}. \emph{Faraday
  Discuss.} \textbf{2016}, \emph{195}, 497--520\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Simm \latin{et~al.}(2017)Simm, Proppe, and Reiher]{Simm2017}
Simm,~G.~N.; Proppe,~J.; Reiher,~M. {Error Assessment of Computational Models
  in Chemistry}. \emph{Chimia} \textbf{2017}, \emph{71}, 202\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Brunken and Reiher(2021)Brunken, and Reiher]{Brunken2021}
Brunken,~C.; Reiher,~M. {Automated Construction of QuantumClassical Hybrid
  Models}. \emph{J. Chem. Theory Comput.} \textbf{2021}, \emph{17},
  3797--3813\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Brunken and Reiher(2020)Brunken, and Reiher]{Brunken2020}
Brunken,~C.; Reiher,~M. {Self-Parametrizing System-Focused Atomistic Models}.
  \emph{J. Chem. Theory Comput.} \textbf{2020}, \emph{16}, 1646--1665\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Alongi and Shields(2010)Alongi, and Shields]{Alongi2010}
Alongi,~K.~S.; Shields,~G.~C. \emph{Annual Reports in Computational Chemistry};
  Elsevier: Amsterdam, The Netherlands, 2010; pp 113--138\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ho and Coote(2009)Ho, and Coote]{Ho2009}
Ho,~J.; Coote,~M.~L. {A universal approach for continuum solvent pKa
  calculations: Are we there yet?} \emph{Theor. Chem. Acc.} \textbf{2009},
  \emph{125}, 3--21\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Casasnovas \latin{et~al.}(2014)Casasnovas, Ortega-Castro, Frau,
  Donoso, and Mu{\~{n}}oz]{Casasnovas2014}
Casasnovas,~R.; Ortega-Castro,~J.; Frau,~J.; Donoso,~J.; Mu{\~{n}}oz,~F.
  {Theoretical pKa calculations with continuum model solvents, alternative
  protocols to thermodynamic cycles}. \textbf{2014}, \emph{114},
  1350--1363\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ho(2015)]{Ho2015}
Ho,~J. {Are thermodynamic cycles necessary for continuum solvent calculation of
  pKas and reduction potentials?} \emph{Phys. Chem. Chem. Phys.} \textbf{2015},
  \emph{17}, 2859--2868\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Seybold and Shields(2015)Seybold, and Shields]{Seybold2015}
Seybold,~P.~G.; Shields,~G.~C. {Computational estimation of pKa values}.
  \emph{Wiley Interdiscip. Rev. Comput. Mol. Sci.} \textbf{2015}, \emph{5},
  290--297\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bashford and Karplus(1990)Bashford, and Karplus]{Bashford1990}
Bashford,~D.; Karplus,~M. {pKa's of Ionizable Groups in Proteins: Atomic Detail
  from a Continuum Electrostatic Model}. \emph{Biochemistry} \textbf{1990},
  \emph{29}, 10219--10225\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wang \latin{et~al.}(2015)Wang, Li, and Alexov]{Wang2015a}
Wang,~L.; Li,~L.; Alexov,~E. {pKa predictions for proteins, RNAs, and DNAs with
  the Gaussian dielectric function using DelPhi pKa}. \emph{Proteins Struct.
  Funct. Bioinforma.} \textbf{2015}, \emph{83}, 2186--2197\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ho(2014)]{Ho2014}
Ho,~J. {Predicting pKa in implicit solvents: Current status and future
  directions}. \emph{{Aust. J. Chem.}} \textbf{2014}, \emph{67},
  1441--1460\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Klici{\'{c}} \latin{et~al.}(2002)Klici{\'{c}}, Friesner, Liu, and
  Guida]{Klicic2002}
Klici{\'{c}},~J.~J.; Friesner,~R.~A.; Liu,~S.~Y.; Guida,~W.~C. {Accurate
  prediction of acidity constants in aqueous solution via density functional
  theory and self-consistent reaction field methods}. \emph{J. Phys. Chem. A}
  \textbf{2002}, \emph{106}, 1327--1335\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Klamt \latin{et~al.}(2003)Klamt, Eckert, Diedenhofen, and
  Beck]{Klamt2003}
Klamt,~A.; Eckert,~F.; Diedenhofen,~M.; Beck,~M.~E. {First Principles
  Calculations of Aqueous pKa Values for Organic and Inorganic Acids Using
  COSMO-RS Reveal an Inconsistency in the Slope of the pKa Scale}. \emph{J.
  Phys. Chem. A} \textbf{2003}, \emph{107}, 9380--9386\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Riojas and Wilson(2014)Riojas, and Wilson]{Riojas2014}
Riojas,~A.~G.; Wilson,~A.~K. {Solv-ccCA: Implicit solvation and the correlation
  consistent composite approach for the determination of p Ka}. \emph{J. Chem.
  Theory Comput.} \textbf{2014}, \emph{10}, 1500--1510\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Thapa and Schlegel(2017)Thapa, and Schlegel]{Thapa2017}
Thapa,~B.; Schlegel,~H.~B. {Improved pKa Prediction of Substituted Alcohols,
  Phenols, and Hydroperoxides in Aqueous Medium Using Density Functional Theory
  and a Cluster-Continuum Solvation Model}. \emph{J. Phys. Chem. A}
  \textbf{2017}, \emph{121}, 4698--4706\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Lian \latin{et~al.}(2018)Lian, Johnston, Parks, and Smith]{Lian2018}
Lian,~P.; Johnston,~R.~C.; Parks,~J.~M.; Smith,~J.~C. {Quantum Chemical
  Calculation of p Kas of Environmentally Relevant Functional Groups:
  Carboxylic Acids, Amines, and Thiols in Aqueous Solution}. \emph{J. Phys.
  Chem. A} \textbf{2018}, \emph{122}, 4366--4374\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Donnini \latin{et~al.}(2011)Donnini, Tegeler, Groenhof, and
  Grubm{\"{u}}ller]{Donnini2011}
Donnini,~S.; Tegeler,~F.; Groenhof,~G.; Grubm{\"{u}}ller,~H. {Constant pH
  Molecular Dynamics in Explicit Solvent with $\lambda$-Dynamics}. \emph{J.
  Chem. Theory Comput.} \textbf{2011}, \emph{7}, 1962--1978\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Itoh \latin{et~al.}(2011)Itoh, Damjanovi{\'{c}}, and Brooks]{Itoh2011}
Itoh,~S.~G.; Damjanovi{\'{c}},~A.; Brooks,~B.~R. {pH replica-exchange method
  based on discrete protonation states}. \emph{Proteins Struct. Funct.
  Bioinforma.} \textbf{2011}, \emph{79}, 3420--3436\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wallace and Shen(2011)Wallace, and Shen]{Wallace2011}
Wallace,~J.~A.; Shen,~J.~K. {Continuous Constant pH Molecular Dynamics in
  Explicit Solvent with pH-Based Replica Exchange}. \emph{J. Chem. Theory
  Comput.} \textbf{2011}, \emph{7}, 2617--2629\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Huang \latin{et~al.}(2016)Huang, Chen, Wallace, and Shen]{Huang2016a}
Huang,~Y.; Chen,~W.; Wallace,~J.~A.; Shen,~J. {All-Atom Continuous Constant pH
  Molecular Dynamics With Particle Mesh Ewald and Titratable Water}. \emph{J.
  Chem. Theory Comput.} \textbf{2016}, \emph{12}, 5411--5421\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li \latin{et~al.}(2002)Li, Hains, Everts, Robertson, and
  Jensen]{Li2002}
Li,~H.; Hains,~A.~W.; Everts,~J.~E.; Robertson,~A.~D.; Jensen,~J.~H. {The
  Prediction of Protein p K a 's Using QM/MM: The p K a of Lysine 55 in Turkey
  Ovomucoid Third Domain}. \emph{J. Phys. Chem. B} \textbf{2002}, \emph{106},
  3486--3494\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li and Cui(2003)Li, and Cui]{Li2003}
Li,~G.; Cui,~Q. {p K a Calculations with QM/MM Free Energy Perturbations}.
  \emph{J. Phys. Chem. B} \textbf{2003}, \emph{107}, 14521--14528\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ho and Coote(2011)Ho, and Coote]{Ho2011}
Ho,~J.; Coote,~M.~L. {Firstprinciples prediction of acidities in the gas and
  solution phase}. \emph{WIREs Comput. Mol. Sci.} \textbf{2011}, \emph{1},
  649--660\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Godoy-Ruiz \latin{et~al.}(2005)Godoy-Ruiz, Perez-Jimenez, Garcia-Mira,
  {Plaza del Pino}, and Sanchez-Ruiz]{Godoy-Ruiz2005}
Godoy-Ruiz,~R.; Perez-Jimenez,~R.; Garcia-Mira,~M.~M.; {Plaza del Pino},~I.~M.;
  Sanchez-Ruiz,~J.~M. Empirical parametrization of pK values for carboxylic
  acids in proteins using a genetic algorithm. \emph{Biophysical Chemistry}
  \textbf{2005}, \emph{115}, 263--266\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li \latin{et~al.}(2005)Li, Robertson, and Jensen]{Li2005}
Li,~H.; Robertson,~A.~D.; Jensen,~J.~H. {Very fast empirical prediction and
  rationalization of protein pK a values}. \emph{Proteins Struct. Funct.
  Genet.} \textbf{2005}, \emph{61}, 704--721\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Krieger \latin{et~al.}(2006)Krieger, Nielsen, Spronk, and
  Vriend]{Krieger2006}
Krieger,~E.; Nielsen,~J.~E.; Spronk,~C.~A.; Vriend,~G. {Fast empirical pKa
  prediction by Ewald summation}. \emph{J. Mol. Graph. Model.} \textbf{2006},
  \emph{25}, 481--486\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bas \latin{et~al.}(2008)Bas, Rogers, and Jensen]{Bas2008b}
Bas,~D.~C.; Rogers,~D.~M.; Jensen,~J.~H. {Very fast prediction and
  rationalization of pKa values for protein-ligand complexes}. \emph{Proteins
  Struct. Funct. Genet.} \textbf{2008}, \emph{73}, 765--783\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Reis \latin{et~al.}(2020)Reis, Vila-Vicosa, Rocchia, and
  Machuqueiro]{Reis2020}
Reis,~P.~B.; Vila-Vicosa,~D.; Rocchia,~W.; Machuqueiro,~M. {PypKA: A flexible
  python module for poisson-Boltzmann-based pKa calculations}. \emph{J. Chem.
  Inf. Model.} \textbf{2020}, \emph{60}, 4442--4448\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dolinsky \latin{et~al.}(2004)Dolinsky, Nielsen, McCammon, and
  Baker]{Dolinsky2004}
Dolinsky,~T.~J.; Nielsen,~J.~E.; McCammon,~J.~A.; Baker,~N.~A. {PDB2PQR: An
  automated pipeline for the setup of Poisson-Boltzmann electrostatics
  calculations}. \emph{Nucleic Acids Res.} \textbf{2004}, \emph{32}\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dolinsky \latin{et~al.}(2007)Dolinsky, Czodrowski, Li, Nielsen,
  Jensen, Klebe, and Baker]{Dolinsky2007}
Dolinsky,~T.~J.; Czodrowski,~P.; Li,~H.; Nielsen,~J.~E.; Jensen,~J.~H.;
  Klebe,~G.; Baker,~N.~A. {PDB2PQR: Expanding and upgrading automated
  preparation of biomolecular structures for molecular simulations}.
  \emph{Nucleic Acids Res.} \textbf{2007}, \emph{35}, W522\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Anandakrishnan \latin{et~al.}(2012)Anandakrishnan, Aguilar, and
  Onufriev]{Anandakrishnan2012}
Anandakrishnan,~R.; Aguilar,~B.; Onufriev,~A.~V. {H++ 3.0: Automating pK
  prediction and the preparation of biomolecular structures for atomistic
  molecular modeling and simulations}. \emph{Nucleic Acids Res.} \textbf{2012},
  \emph{40}, 537--541\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Olsson \latin{et~al.}(2011)Olsson, S{\o}ndergaard, Rostkowski, and
  Jensen]{Olsson2011}
Olsson,~M.~H.; S{\o}ndergaard,~C.~R.; Rostkowski,~M.; Jensen,~J.~H. {PROPKA3:
  Consistent treatment of internal and surface residues in empirical p K a
  predictions}. \emph{J. Chem. Theory Comput.} \textbf{2011}, \emph{7},
  525--537\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bochevarov \latin{et~al.}(2016)Bochevarov, Watson, Greenwood, and
  Philipp]{Bochevarov2016a}
Bochevarov,~A.~D.; Watson,~M.~A.; Greenwood,~J.~R.; Philipp,~D.~M.
  {Multiconformation, Density Functional Theory-Based pKa Prediction in
  Application to Large, Flexible Organic Molecules with Diverse Functional
  Groups}. \emph{J. Chem. Theory Comput.} \textbf{2016}, \emph{12},
  6001--6019\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ratkova \latin{et~al.}(2015)Ratkova, Palmer, and Fedorov]{Ratkova2015}
Ratkova,~E.~L.; Palmer,~D.~S.; Fedorov,~M.~V. {Solvation Thermodynamics of
  Organic Molecules by the Molecular Integral Equation Theory: Approaching
  Chemical Accuracy}. \emph{Chem. Rev.} \textbf{2015}, \emph{115},
  6312--6356\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Matsui \latin{et~al.}(2012)Matsui, Baba, Kamiya, and
  Shigeta]{Matsui2012}
Matsui,~T.; Baba,~T.; Kamiya,~K.; Shigeta,~Y. {An accurate density functional
  theory based estimation of pK a values of polar residues combined with
  experimental data: From amino acids to minimal proteins}. \emph{Phys. Chem.
  Chem. Phys.} \textbf{2012}, \emph{14}, 4181--4187\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Pernot \latin{et~al.}(2015)Pernot, Civalleri, Presti, and
  Savin]{Pernot2015}
Pernot,~P.; Civalleri,~B.; Presti,~D.; Savin,~A. {Prediction Uncertainty of
  Density Functional Approximations for Properties of Crystals with Cubic
  Symmetry}. \emph{J. Phys. Chem. A} \textbf{2015}, \emph{119},
  5288--5304\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Proppe and Reiher(2017)Proppe, and Reiher]{Proppe2017}
Proppe,~J.; Reiher,~M. {Reliable Estimation of Prediction Uncertainty for
  Physicochemical Property Models}. \emph{J. Chem. Theory Comput.}
  \textbf{2017}, \emph{13}, 3297--3317\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Simm and Reiher(2016)Simm, and Reiher]{Simm2016}
Simm,~G.~N.; Reiher,~M. {Systematic Error Estimation for Chemical Reaction
  Energies}. \emph{J. Chem. Theory Comput.} \textbf{2016}, \emph{12},
  2762--2773\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Proppe and Reiher(2019)Proppe, and Reiher]{Proppe2019}
Proppe,~J.; Reiher,~M. {Mechanism Deduction from Noisy Chemical Reaction
  Networks}. \emph{J. Chem. Theory Comput.} \textbf{2019}, \emph{15},
  357--370\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Reiher(2022)]{Reiher2021}
Reiher,~M. {MoleculeSpecific Uncertainty Quantification in Quantum Chemical
  Studies}. \emph{Isr. J. Chem.} \textbf{2022}, \emph{62}, 1--34\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Simm and Reiher(2018)Simm, and Reiher]{Simm2018}
Simm,~G.~N.; Reiher,~M. {Error-Controlled Exploration of Chemical Reaction
  Networks with Gaussian Processes}. \emph{J. Chem. Theory Comput.}
  \textbf{2018}, \emph{14}, 5238--5248\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Proppe \latin{et~al.}(2019)Proppe, Gugler, and Reiher]{Proppe2019a}
Proppe,~J.; Gugler,~S.; Reiher,~M. {Gaussian Process-Based Refinement of
  Dispersion Corrections}. \emph{J. Chem. Theory Comput.} \textbf{2019},
  \emph{15}, 6046--6060\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Deringer \latin{et~al.}(2021)Deringer, Bart{\'{o}}k, Bernstein,
  Wilkins, Ceriotti, and Cs{\'{a}}nyi]{Deringer2021a}
Deringer,~V.~L.; Bart{\'{o}}k,~A.~P.; Bernstein,~N.; Wilkins,~D.~M.;
  Ceriotti,~M.; Cs{\'{a}}nyi,~G. {Gaussian Process Regression for Materials and
  Molecules}. \emph{{Chem. Rev.}} \textbf{2021}, \emph{121}, 10073--10141\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Gugler and Reiher(2022)Gugler, and Reiher]{Gugler2022}
Gugler,~S.; Reiher,~M. {Quantum Chemical Roots of Machine-Learning Molecular
  Similarity Descriptors}. \emph{J. Chem. Theory Comput.} \textbf{2022},
  \emph{18}, 6670--6689\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Neese(2012)]{Neese2012}
Neese,~F. {The ORCA program system}. \emph{Wiley Int. Rev. Comput. Mol. Sci.}
  \textbf{2012}, \emph{2}, 73--78\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Neese \latin{et~al.}(2020)Neese, Wennmohs, Becker, and
  Riplinger]{Neese2020}
Neese,~F.; Wennmohs,~F.; Becker,~U.; Riplinger,~C. {The ORCA quantum chemistry
  program package}. \emph{J. Chem. Phys.} \textbf{2020}, \emph{152},
  224108\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Perdew \latin{et~al.}(1996)Perdew, Burke, and Ernzerhof]{Perdew1996a}
Perdew,~J.~P.; Burke,~K.; Ernzerhof,~M. {Generalized gradient approximation
  made simple}. \emph{Phys. Rev. Lett.} \textbf{1996}, \emph{77},
  3865--3868\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ernzerhof and Scuseria(1999)Ernzerhof, and Scuseria]{Ernzerhof1999}
Ernzerhof,~M.; Scuseria,~G.~E. {Assessment of the Perdew-Burke-Ernzerhof
  exchange-correlation functional}. \emph{J. Chem. Phys.} \textbf{1999},
  \emph{110}, 5029--5036\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Adamo and Barone(1999)Adamo, and Barone]{Adamo1999}
Adamo,~C.; Barone,~V. {Toward reliable density functional methods without
  adjustable parameters: The PBE0 model}. \emph{J. Chem. Phys.} \textbf{1999},
  \emph{110}, 6158--6170\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Sch{\"{a}}fer \latin{et~al.}(1994)Sch{\"{a}}fer, Huber, and
  Ahlrichs]{Schafer1994}
Sch{\"{a}}fer,~A.; Huber,~C.; Ahlrichs,~R. {Fully optimized contracted Gaussian
  basis sets of triple zeta valence quality for atoms Li to Kr}. \emph{J. Chem.
  Phys.} \textbf{1994}, \emph{100}, 5829--5835\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Eichkorn \latin{et~al.}(1995)Eichkorn, Treutler, {\"{O}}hm,
  H{\"{a}}ser, and Ahlrichs]{Eichkorn1995}
Eichkorn,~K.; Treutler,~O.; {\"{O}}hm,~H.; H{\"{a}}ser,~M.; Ahlrichs,~R.
  {Auxiliary basis sets to approximate Coulomb potentials}. \emph{Chem. Phys.
  Lett.} \textbf{1995}, \emph{240}, 283--290\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Eichkorn \latin{et~al.}(1997)Eichkorn, Weigend, Treutler, and
  Ahlrichs]{Eichkorn1997}
Eichkorn,~K.; Weigend,~F.; Treutler,~O.; Ahlrichs,~R. {Auxiliary basis sets for
  main row atoms and transition metals and their use to approximate Coulomb
  potentials}. \emph{Theor. Chem. Acc.} \textbf{1997}, \emph{97},
  119--124\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Grimme \latin{et~al.}(2010)Grimme, Antony, Ehrlich, and
  Krieg]{Grimme2010a}
Grimme,~S.; Antony,~J.; Ehrlich,~S.; Krieg,~H. {A consistent and accurate ab
  initio parametrization of density functional dispersion correction (DFT-D)
  for the 94 elements H-Pu}. \emph{J. Chem. Phys.} \textbf{2010}, \emph{132},
  154104\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Grimme \latin{et~al.}(2011)Grimme, Ehrlich, and Goerigk]{Grimme2011}
Grimme,~S.; Ehrlich,~S.; Goerigk,~L. {Effect of the damping function in
  dispersion corrected density functional theory}. \emph{J. Comput. Chem.}
  \textbf{2011}, \emph{32}, 1456--1465\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Marenich \latin{et~al.}(2009)Marenich, Cramer, and
  Truhlar]{Marenich2009}
Marenich,~A.~V.; Cramer,~C.~J.; Truhlar,~D.~G. {Universal Solvation Model Based
  on Solute Electron Density and on a Continuum Model of the Solvent Defined by
  the Bulk Dielectric Constant and Atomic Surface Tensions}. \emph{J. Phys.
  Chem. B} \textbf{2009}, \emph{113}, 6378--6396\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Sobez and Reiher(2020)Sobez, and Reiher]{Sobez2020}
Sobez,~J.~G.; Reiher,~M. {Molassembler: Molecular Graph Construction,
  Modification, and Conformer Generation for Inorganic and Organic Molecules}.
  \emph{J. Chem. Inf. Model.} \textbf{2020}, \emph{60}, 3884--3900\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bensberg \latin{et~al.}(2023)Bensberg, Grimmel, Sobez, Steiner,
  Unsleber, and Reiher]{Molassembler2023}
Bensberg,~M.; Grimmel,~S.~A.; Sobez,~J.-G.; Steiner,~M.; Unsleber,~J.~P.;
  Reiher,~M. {qcscine/Molassembler: Release 2.0.0}. 2023;
  \url{https://zenodo.org/record/7928074}\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Baiardi \latin{et~al.}(2023)Baiardi, Bensberg, Bosia, Brunken, Csizi,
  Feldmann, Glaser, Grimmel, Gugler, Haag, Heuer, M{\"u}ller, Polonius, Simm,
  Sobez, Steiner, T{\"u}rtscher, Unsleber, Vaucher, Weymuth, and
  Reiher]{Utils2023}
Baiardi,~A.; Bensberg,~M.; Bosia,~F.; Brunken,~C.; Csizi,~K.-S.; Feldmann,~R.;
  Glaser,~N.; Grimmel,~S.~A.; Gugler,~S.; Haag,~M.~P.; Heuer,~M.~A.;
  M{\"u}ller,~C.~H.; Polonius,~S.; Simm,~G.~N.; Sobez,~J.-G.; Steiner,~M.;
  T{\"u}rtscher,~P.~L.; Unsleber,~J.~P.; Vaucher,~A.~C.; Weymuth,~T.;
  Reiher,~M. {qcscine/Utilities: Release 8.0.0}. 2023;
  \url{https://zenodo.org/record/7928050}\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[{The PyMOL Molecular Graphics System, Version 2.0
  Schr{\"{o}}dinger}(2015)]{PyMOL}
{The PyMOL Molecular Graphics System, Version 2.0 Schr{\"{o}}dinger},~L.
  {https://pymol.org (accessed November 14, 2022) -- The PyMOL Molecular
  Graphics System, Version 2.5.2}. 2015\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Artymiuk \latin{et~al.}(1994)Artymiuk, Poirrette, Grindley, Rice, and
  Willett]{Artymiuk1994}
Artymiuk,~P.~J.; Poirrette,~A.~R.; Grindley,~H.~M.; Rice,~D.~W.; Willett,~P. {A
  graph-theoretic approach to the identification of three-dimensional patterns
  of amino acid side-chains in protein structures}. \emph{J. Mol. Biol.}
  \textbf{1994}, \emph{243}, 327--344\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Spriggs \latin{et~al.}(2003)Spriggs, Artymiuk, and
  Willett]{Spriggs2003}
Spriggs,~R.~V.; Artymiuk,~P.~J.; Willett,~P. {Searching for patterns of amino
  acids in 3D protein structures}. \emph{{J. Chem. Inf. Comput. Sci.}}
  \textbf{2003}, \emph{43}, 412--421\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Simm \latin{et~al.}(2020)Simm, T{\"{u}}rtscher, and Reiher]{Simm2020}
Simm,~G.~N.; T{\"{u}}rtscher,~P.~L.; Reiher,~M. {Systematic microsolvation
  approach with a clustercontinuum scheme and conformational sampling}.
  \emph{J. Comput. Chem.} \textbf{2020}, \emph{41}, 1144--1155\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Xue \latin{et~al.}(1998)Xue, {\"{O}}kvist, Hansson, and
  Young]{Xue1998}
Xue,~Y.; {\"{O}}kvist,~M.; Hansson,~{\"{O}}.; Young,~S. {Crystal structure of
  spinach plastocyanin at 1.7 \AA resolution}. \emph{Protein Sci.}
  \textbf{1998}, \emph{7}, 2099--2105\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[{IUPAC-IUB Commission on Biochemical
  Nomenclature}(1970)]{iupacAtomtyping}
{IUPAC-IUB Commission on Biochemical Nomenclature}, Abbreviations and Symbols
  for the Description of the Conformation of Polypeptide Chains: Tentative
  Rules (1969). \emph{Journal of Biological Chemistry} \textbf{1970},
  \emph{245}, 6489--6497\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[O'Boyle \latin{et~al.}(2011)O'Boyle, Banck, James, Morley,
  Vandermeersch, and Hutchison]{OpenBabel}
O'Boyle,~N.~M.; Banck,~M.; James,~C.~A.; Morley,~C.; Vandermeersch,~T.;
  Hutchison,~G.~R. {Open Babel: An open chemical toolbox}. \emph{J.
  Cheminform.} \textbf{2011}, \emph{3}, 33\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bensberg \latin{et~al.}(2022)Bensberg, Grimmel, Sobez, Steiner,
  Unsleber, and Reiher]{database1.0.0}
Bensberg,~M.; Grimmel,~S.~A.; Sobez,~J.-G.; Steiner,~M.; Unsleber,~J.~P.;
  Reiher,~M. {qcscine/database: Release 1.0.0}. 2022\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Chang \latin{et~al.}(1997)Chang, J{\o}rgensen, Bardrum, and
  Led]{Chang1997}
Chang,~X.; J{\o}rgensen,~A. M.~M.; Bardrum,~P.; Led,~J.~J. {Solution structures
  of the R6 human insulin hexamer}. \emph{Biochemistry} \textbf{1997},
  \emph{36}, 9409--9422\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Karlin and Zhu(1997)Karlin, and Zhu]{Zhu1997}
Karlin,~S.; Zhu,~Z.-Y. {Classification of mononuclear zinc metal sites in
  protein structures}. \emph{Proc. Natl. Acad. Sci.} \textbf{1997}, \emph{94},
  14231--14236\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Rost and Sander(1993)Rost, and Sander]{Rost1993}
Rost,~B.; Sander,~C. {Prediction of protein secondary structure at better than
  70\% accuracy}. \emph{J. Mol. Biol.} \textbf{1993}, \emph{232},
  584--599\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Cuff and Barton(1999)Cuff, and Barton]{Cuff1999}
Cuff,~J.~A.; Barton,~G.~J. {Evaluation and improvement of multiple sequence
  methods for protein secondary structure prediction}. \emph{Proteins Struct.
  Funct. Genet.} \textbf{1999}, \emph{34}, 508--519\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Rost and Eyrich(2001)Rost, and Eyrich]{Rost2001}
Rost,~B.; Eyrich,~V.~A. {EVA: Large-scale analysis of secondary structure
  prediction}. \emph{Proteins Struct. Funct. Genet.} \textbf{2001}, \emph{45},
  192--199\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Thapa and Schlegel(2017)Thapa, and Schlegel]{Thapa2017a}
Thapa,~B.; Schlegel,~H.~B. {Improved pKa Prediction of Substituted Alcohols,
  Phenols, and Hydroperoxides in Aqueous Medium Using Density Functional Theory
  and a Cluster-Continuum Solvation Model}. \emph{J. Phys. Chem. A}
  \textbf{2017}, \emph{121}, 4698--4706\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Thapa and Schlegel(2016)Thapa, and Schlegel]{Thapa2016}
Thapa,~B.; Schlegel,~H.~B. {Density Functional Theory Calculation of pKa's of
  Thiols in Aqueous Solution Using Explicit Water Molecules and the Polarizable
  Continuum Model}. \emph{J. Phys. Chem. A} \textbf{2016}, \emph{120},
  5726--5735\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[{Schmidt Am Busch} and Knapp(2004){Schmidt Am Busch}, and
  Knapp]{SchmidtAmBusch2004}
{Schmidt Am Busch},~M.; Knapp,~E.~W. {Accurate pKa determination for a
  heterogeneous group of organic molecules}. \emph{ChemPhysChem} \textbf{2004},
  \emph{5}, 1513--1522\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Jurani{\'{c}}(2014)]{Juranic2014}
Jurani{\'{c}},~I. {Simple method for the estimation of pKa of amines}.
  \emph{Croat. Chem. Acta} \textbf{2014}, \emph{87}, 343--347\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Beale \latin{et~al.}(2013)Beale, Chudzinski, Sarwar, and
  Taylor]{Beale2013}
Beale,~T.~M.; Chudzinski,~M.~G.; Sarwar,~M.~G.; Taylor,~M.~S. {Halogen bonding
  in solution: thermodynamics and applications}. \emph{Chem. Soc. Rev.}
  \textbf{2013}, \emph{42}, 1667--1680\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Kromann \latin{et~al.}(2016)Kromann, Larsen, Moustafa, and
  Jensen]{Kromann2016}
Kromann,~J.~C.; Larsen,~F.; Moustafa,~H.; Jensen,~J.~H. {Prediction of pKa
  values using the PM6 semiempirical method}. \emph{PeerJ} \textbf{2016},
  \emph{2016}, e2335\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Jensen \latin{et~al.}(2017)Jensen, Swain, and Olsen]{Jensen2017}
Jensen,~J.~H.; Swain,~C.~J.; Olsen,~L. {Prediction of p K a Values for Druglike
  Molecules Using Semiempirical Quantum Chemical Methods}. \emph{J. Phys. Chem.
  A} \textbf{2017}, \emph{121}, 699--707\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Hunt \latin{et~al.}(2020)Hunt, Hosseini-Gerami, Chrien, Plante,
  Ponting, and Segall]{Hunt2020}
Hunt,~P.; Hosseini-Gerami,~L.; Chrien,~T.; Plante,~J.; Ponting,~D.~J.;
  Segall,~M. {Predicting p K a Using a Combination of Semi-Empirical Quantum
  Mechanics and Radial Basis Function Methods}. \emph{J. Chem. Inf. Model.}
  \textbf{2020}, \emph{60}, 2989--2997\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[lit()]{litPkasAA}
pKa values of amino acids at 25 C.
  \url{https://www.vanderbilt.edu/AnS/Chemistry/Rizzo/stuff/AA/AminoAcids.html},
  Accessed: 2022-09-14\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\end{mcitethebibliography}

%\clearpage 
%
%\begin{center}
%\textbf{Table-of-Contents Figure}
%% Figure environment removed
%\end{center}

\end{document}
